[2015-05-30 16:54:54,063] INFO Verifying properties (kafka.utils.VerifiableProperties)
[2015-05-30 16:54:54,094] INFO Property broker.id is overridden to 0 (kafka.utils.VerifiableProperties)
[2015-05-30 16:54:54,094] INFO Property log.cleaner.enable is overridden to false (kafka.utils.VerifiableProperties)
[2015-05-30 16:54:54,094] INFO Property log.dirs is overridden to /tmp/kafka-logs (kafka.utils.VerifiableProperties)
[2015-05-30 16:54:54,094] INFO Property log.retention.check.interval.ms is overridden to 300000 (kafka.utils.VerifiableProperties)
[2015-05-30 16:54:54,094] INFO Property log.retention.hours is overridden to 168 (kafka.utils.VerifiableProperties)
[2015-05-30 16:54:54,094] INFO Property log.segment.bytes is overridden to 1073741824 (kafka.utils.VerifiableProperties)
[2015-05-30 16:54:54,094] INFO Property num.io.threads is overridden to 8 (kafka.utils.VerifiableProperties)
[2015-05-30 16:54:54,094] INFO Property num.network.threads is overridden to 3 (kafka.utils.VerifiableProperties)
[2015-05-30 16:54:54,094] INFO Property num.partitions is overridden to 1 (kafka.utils.VerifiableProperties)
[2015-05-30 16:54:54,094] INFO Property num.recovery.threads.per.data.dir is overridden to 1 (kafka.utils.VerifiableProperties)
[2015-05-30 16:54:54,094] INFO Property port is overridden to 9092 (kafka.utils.VerifiableProperties)
[2015-05-30 16:54:54,094] INFO Property socket.receive.buffer.bytes is overridden to 102400 (kafka.utils.VerifiableProperties)
[2015-05-30 16:54:54,094] INFO Property socket.request.max.bytes is overridden to 104857600 (kafka.utils.VerifiableProperties)
[2015-05-30 16:54:54,094] INFO Property socket.send.buffer.bytes is overridden to 102400 (kafka.utils.VerifiableProperties)
[2015-05-30 16:54:54,094] INFO Property zookeeper.connect is overridden to localhost:2181 (kafka.utils.VerifiableProperties)
[2015-05-30 16:54:54,094] INFO Property zookeeper.connection.timeout.ms is overridden to 6000 (kafka.utils.VerifiableProperties)
[2015-05-30 16:54:54,141] INFO [Kafka Server 0], starting (kafka.server.KafkaServer)
[2015-05-30 16:54:54,141] INFO [Kafka Server 0], Connecting to zookeeper on localhost:2181 (kafka.server.KafkaServer)
[2015-05-30 16:55:01,891] FATAL [Kafka Server 0], Fatal error during KafkaServer startup. Prepare to shutdown (kafka.server.KafkaServer)
org.I0Itec.zkclient.exception.ZkTimeoutException: Unable to connect to zookeeper server within timeout: 6000
	at org.I0Itec.zkclient.ZkClient.connect(ZkClient.java:880)
	at org.I0Itec.zkclient.ZkClient.<init>(ZkClient.java:98)
	at org.I0Itec.zkclient.ZkClient.<init>(ZkClient.java:84)
	at kafka.server.KafkaServer.initZk(KafkaServer.scala:157)
	at kafka.server.KafkaServer.startup(KafkaServer.scala:82)
	at kafka.server.KafkaServerStartable.startup(KafkaServerStartable.scala:29)
	at kafka.Kafka$.main(Kafka.scala:46)
	at kafka.Kafka.main(Kafka.scala)
[2015-05-30 16:55:01,891] INFO [Kafka Server 0], shutting down (kafka.server.KafkaServer)
[2015-05-30 16:55:01,901] INFO [Kafka Server 0], shut down completed (kafka.server.KafkaServer)
[2015-05-30 16:55:01,901] FATAL Fatal error during KafkaServerStartable startup. Prepare to shutdown (kafka.server.KafkaServerStartable)
org.I0Itec.zkclient.exception.ZkTimeoutException: Unable to connect to zookeeper server within timeout: 6000
	at org.I0Itec.zkclient.ZkClient.connect(ZkClient.java:880)
	at org.I0Itec.zkclient.ZkClient.<init>(ZkClient.java:98)
	at org.I0Itec.zkclient.ZkClient.<init>(ZkClient.java:84)
	at kafka.server.KafkaServer.initZk(KafkaServer.scala:157)
	at kafka.server.KafkaServer.startup(KafkaServer.scala:82)
	at kafka.server.KafkaServerStartable.startup(KafkaServerStartable.scala:29)
	at kafka.Kafka$.main(Kafka.scala:46)
	at kafka.Kafka.main(Kafka.scala)
[2015-05-30 16:55:01,901] INFO [Kafka Server 0], shutting down (kafka.server.KafkaServer)
[2015-05-30 16:55:08,134] INFO Verifying properties (kafka.utils.VerifiableProperties)
[2015-05-30 16:55:08,166] INFO Property broker.id is overridden to 0 (kafka.utils.VerifiableProperties)
[2015-05-30 16:55:08,166] INFO Property log.cleaner.enable is overridden to false (kafka.utils.VerifiableProperties)
[2015-05-30 16:55:08,166] INFO Property log.dirs is overridden to /tmp/kafka-logs (kafka.utils.VerifiableProperties)
[2015-05-30 16:55:08,166] INFO Property log.retention.check.interval.ms is overridden to 300000 (kafka.utils.VerifiableProperties)
[2015-05-30 16:55:08,166] INFO Property log.retention.hours is overridden to 168 (kafka.utils.VerifiableProperties)
[2015-05-30 16:55:08,166] INFO Property log.segment.bytes is overridden to 1073741824 (kafka.utils.VerifiableProperties)
[2015-05-30 16:55:08,166] INFO Property num.io.threads is overridden to 8 (kafka.utils.VerifiableProperties)
[2015-05-30 16:55:08,166] INFO Property num.network.threads is overridden to 3 (kafka.utils.VerifiableProperties)
[2015-05-30 16:55:08,166] INFO Property num.partitions is overridden to 1 (kafka.utils.VerifiableProperties)
[2015-05-30 16:55:08,166] INFO Property num.recovery.threads.per.data.dir is overridden to 1 (kafka.utils.VerifiableProperties)
[2015-05-30 16:55:08,166] INFO Property port is overridden to 9092 (kafka.utils.VerifiableProperties)
[2015-05-30 16:55:08,166] INFO Property socket.receive.buffer.bytes is overridden to 102400 (kafka.utils.VerifiableProperties)
[2015-05-30 16:55:08,166] INFO Property socket.request.max.bytes is overridden to 104857600 (kafka.utils.VerifiableProperties)
[2015-05-30 16:55:08,166] INFO Property socket.send.buffer.bytes is overridden to 102400 (kafka.utils.VerifiableProperties)
[2015-05-30 16:55:08,166] INFO Property zookeeper.connect is overridden to localhost:2181 (kafka.utils.VerifiableProperties)
[2015-05-30 16:55:08,166] INFO Property zookeeper.connection.timeout.ms is overridden to 6000 (kafka.utils.VerifiableProperties)
[2015-05-30 16:55:08,197] INFO [Kafka Server 0], starting (kafka.server.KafkaServer)
[2015-05-30 16:55:08,197] INFO [Kafka Server 0], Connecting to zookeeper on localhost:2181 (kafka.server.KafkaServer)
[2015-05-30 16:55:08,478] INFO Loading logs. (kafka.log.LogManager)
[2015-05-30 16:55:08,525] INFO Recovering unflushed segment 0 in log my-repilcated-topic-0. (kafka.log.Log)
[2015-05-30 16:55:08,525] WARN Error when freeing index buffer (kafka.log.OffsetIndex)
java.lang.NullPointerException
	at kafka.log.OffsetIndex.kafka$log$OffsetIndex$$forceUnmap(OffsetIndex.scala:301)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:283)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:276)
	at kafka.utils.Utils$.inLock(Utils.scala:535)
	at kafka.log.OffsetIndex.resize(OffsetIndex.scala:276)
	at kafka.log.Log.loadSegments(Log.scala:179)
	at kafka.log.Log.<init>(Log.scala:67)
	at kafka.log.LogManager$$anonfun$loadLogs$2$$anonfun$3$$anonfun$apply$7$$anonfun$apply$1.apply$mcV$sp(LogManager.scala:142)
	at kafka.utils.Utils$$anon$1.run(Utils.scala:54)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 16:55:08,541] INFO Completed load of log my-repilcated-topic-0 with log end offset 19 (kafka.log.Log)
[2015-05-30 16:55:08,556] INFO Recovering unflushed segment 0 in log test-0. (kafka.log.Log)
[2015-05-30 16:55:08,572] WARN Error when freeing index buffer (kafka.log.OffsetIndex)
java.lang.NullPointerException
	at kafka.log.OffsetIndex.kafka$log$OffsetIndex$$forceUnmap(OffsetIndex.scala:301)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:283)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:276)
	at kafka.utils.Utils$.inLock(Utils.scala:535)
	at kafka.log.OffsetIndex.resize(OffsetIndex.scala:276)
	at kafka.log.Log.loadSegments(Log.scala:179)
	at kafka.log.Log.<init>(Log.scala:67)
	at kafka.log.LogManager$$anonfun$loadLogs$2$$anonfun$3$$anonfun$apply$7$$anonfun$apply$1.apply$mcV$sp(LogManager.scala:142)
	at kafka.utils.Utils$$anon$1.run(Utils.scala:54)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 16:55:08,587] INFO Completed load of log test-0 with log end offset 137 (kafka.log.Log)
[2015-05-30 16:55:08,587] INFO Logs loading complete. (kafka.log.LogManager)
[2015-05-30 16:55:08,587] INFO Starting log cleanup with a period of 300000 ms. (kafka.log.LogManager)
[2015-05-30 16:55:08,587] INFO Starting log flusher with a default period of 9223372036854775807 ms. (kafka.log.LogManager)
[2015-05-30 16:55:08,619] INFO Awaiting socket connections on 0.0.0.0:9092. (kafka.network.Acceptor)
[2015-05-30 16:55:08,619] INFO [Socket Server on Broker 0], Started (kafka.network.SocketServer)
[2015-05-30 16:55:08,681] INFO Will not load MX4J, mx4j-tools.jar is not in the classpath (kafka.utils.Mx4jLoader$)
[2015-05-30 16:55:08,712] INFO 0 successfully elected as leader (kafka.server.ZookeeperLeaderElector)
[2015-05-30 16:55:08,869] INFO New leader is 0 (kafka.server.ZookeeperLeaderElector$LeaderChangeListener)
[2015-05-30 16:55:08,884] INFO Registered broker 0 at path /brokers/ids/0 with address MININT-4IL6B3T.fareast.corp.microsoft.com:9092. (kafka.utils.ZkUtils$)
[2015-05-30 16:55:08,900] INFO [Kafka Server 0], started (kafka.server.KafkaServer)
[2015-05-30 16:55:09,072] INFO [ReplicaFetcherManager on broker 0] Removed fetcher for partitions [test,0] (kafka.server.ReplicaFetcherManager)
[2015-05-30 16:55:09,087] INFO [ReplicaFetcherManager on broker 0] Removed fetcher for partitions  (kafka.server.ReplicaFetcherManager)
[2015-05-30 16:55:09,103] INFO [ReplicaFetcherManager on broker 0] Added fetcher for partitions List() (kafka.server.ReplicaFetcherManager)
[2015-05-30 16:55:09,119] INFO [ReplicaFetcherManager on broker 0] Removed fetcher for partitions [my-repilcated-topic,0],[test,0] (kafka.server.ReplicaFetcherManager)
[2015-05-30 16:57:25,768] INFO [ReplicaFetcherManager on broker 0] Removed fetcher for partitions [kafkatopic,0] (kafka.server.ReplicaFetcherManager)
[2015-05-30 16:57:25,783] INFO Completed load of log kafkatopic-0 with log end offset 0 (kafka.log.Log)
[2015-05-30 16:57:25,783] INFO Created log for partition [kafkatopic,0] in D:\tmp\kafka-logs with properties {segment.index.bytes -> 10485760, file.delete.delay.ms -> 60000, segment.bytes -> 1073741824, flush.ms -> 9223372036854775807, delete.retention.ms -> 86400000, index.interval.bytes -> 4096, retention.bytes -> -1, min.insync.replicas -> 1, cleanup.policy -> delete, unclean.leader.election.enable -> true, segment.ms -> 604800000, max.message.bytes -> 1000012, flush.messages -> 9223372036854775807, min.cleanable.dirty.ratio -> 0.5, retention.ms -> 604800000, segment.jitter.ms -> 0}. (kafka.log.LogManager)
[2015-05-30 16:57:25,783] WARN Partition [kafkatopic,0] on broker 0: No checkpointed highwatermark is found for partition [kafkatopic,0] (kafka.cluster.Partition)
[2015-05-30 16:58:54,209] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 17:04:38,773] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 17:04:43,048] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 17:04:43,058] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 17:05:15,388] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at kafka.utils.Utils$.read(Utils.scala:380)
	at kafka.network.BoundedByteBufferReceive.readFrom(BoundedByteBufferReceive.scala:54)
	at kafka.network.Processor.read(SocketServer.scala:444)
	at kafka.network.Processor.run(SocketServer.scala:340)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 17:05:17,824] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:229)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 18:46:32,164] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 18:46:37,223] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 18:47:25,482] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 18:47:25,560] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 18:48:36,306] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 18:48:53,784] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 18:49:51,864] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 18:52:49,785] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at kafka.utils.Utils$.read(Utils.scala:380)
	at kafka.network.BoundedByteBufferReceive.readFrom(BoundedByteBufferReceive.scala:54)
	at kafka.network.Processor.read(SocketServer.scala:444)
	at kafka.network.Processor.run(SocketServer.scala:340)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 18:59:46,111] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 18:59:47,052] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:00:12,588] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 19:00:13,433] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:00:31,697] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 19:00:32,621] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:03:36,155] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 19:03:41,033] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:04:01,354] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 19:04:11,234] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:04:24,230] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:229)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:08:09,174] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 19:08:13,915] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:08:18,878] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:08:30,928] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 19:08:44,618] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:08:53,232] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at kafka.utils.Utils$.read(Utils.scala:380)
	at kafka.network.BoundedByteBufferReceive.readFrom(BoundedByteBufferReceive.scala:54)
	at kafka.network.Processor.read(SocketServer.scala:444)
	at kafka.network.Processor.run(SocketServer.scala:340)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:09:04,900] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:09:13,473] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 19:09:23,357] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:09:26,890] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:229)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:11:45,458] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at kafka.utils.Utils$.read(Utils.scala:380)
	at kafka.network.BoundedByteBufferReceive.readFrom(BoundedByteBufferReceive.scala:54)
	at kafka.network.Processor.read(SocketServer.scala:444)
	at kafka.network.Processor.run(SocketServer.scala:340)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:11:53,143] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:11:56,092] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 19:12:05,924] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:12:10,835] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:229)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:14:22,077] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 19:14:30,005] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at kafka.utils.Utils$.read(Utils.scala:380)
	at kafka.network.BoundedByteBufferReceive.readFrom(BoundedByteBufferReceive.scala:54)
	at kafka.network.Processor.read(SocketServer.scala:444)
	at kafka.network.Processor.run(SocketServer.scala:340)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:14:36,913] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 19:14:42,332] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at kafka.utils.Utils$.read(Utils.scala:380)
	at kafka.network.BoundedByteBufferReceive.readFrom(BoundedByteBufferReceive.scala:54)
	at kafka.network.Processor.read(SocketServer.scala:444)
	at kafka.network.Processor.run(SocketServer.scala:340)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:15:56,455] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 19:16:04,733] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:16:08,162] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at kafka.utils.Utils$.read(Utils.scala:380)
	at kafka.network.BoundedByteBufferReceive.readFrom(BoundedByteBufferReceive.scala:54)
	at kafka.network.Processor.read(SocketServer.scala:444)
	at kafka.network.Processor.run(SocketServer.scala:340)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:16:20,474] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:16:38,419] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 19:18:55,277] INFO Verifying properties (kafka.utils.VerifiableProperties)
[2015-05-30 19:18:55,310] INFO Property broker.id is overridden to 0 (kafka.utils.VerifiableProperties)
[2015-05-30 19:18:55,311] INFO Property log.cleaner.enable is overridden to false (kafka.utils.VerifiableProperties)
[2015-05-30 19:18:55,311] INFO Property log.dirs is overridden to /tmp/kafka-logs (kafka.utils.VerifiableProperties)
[2015-05-30 19:18:55,311] INFO Property log.retention.check.interval.ms is overridden to 300000 (kafka.utils.VerifiableProperties)
[2015-05-30 19:18:55,311] INFO Property log.retention.hours is overridden to 168 (kafka.utils.VerifiableProperties)
[2015-05-30 19:18:55,312] INFO Property log.segment.bytes is overridden to 1073741824 (kafka.utils.VerifiableProperties)
[2015-05-30 19:18:55,312] INFO Property num.io.threads is overridden to 8 (kafka.utils.VerifiableProperties)
[2015-05-30 19:18:55,312] INFO Property num.network.threads is overridden to 3 (kafka.utils.VerifiableProperties)
[2015-05-30 19:18:55,313] INFO Property num.partitions is overridden to 1 (kafka.utils.VerifiableProperties)
[2015-05-30 19:18:55,313] INFO Property num.recovery.threads.per.data.dir is overridden to 1 (kafka.utils.VerifiableProperties)
[2015-05-30 19:18:55,313] INFO Property port is overridden to 9092 (kafka.utils.VerifiableProperties)
[2015-05-30 19:18:55,314] INFO Property socket.receive.buffer.bytes is overridden to 102400 (kafka.utils.VerifiableProperties)
[2015-05-30 19:18:55,314] INFO Property socket.request.max.bytes is overridden to 104857600 (kafka.utils.VerifiableProperties)
[2015-05-30 19:18:55,315] INFO Property socket.send.buffer.bytes is overridden to 102400 (kafka.utils.VerifiableProperties)
[2015-05-30 19:18:55,316] INFO Property zookeeper.connect is overridden to localhost:2181 (kafka.utils.VerifiableProperties)
[2015-05-30 19:18:55,316] INFO Property zookeeper.connection.timeout.ms is overridden to 6000 (kafka.utils.VerifiableProperties)
[2015-05-30 19:18:55,358] INFO [Kafka Server 0], starting (kafka.server.KafkaServer)
[2015-05-30 19:18:55,361] INFO [Kafka Server 0], Connecting to zookeeper on localhost:2181 (kafka.server.KafkaServer)
[2015-05-30 19:18:55,661] INFO Loading logs. (kafka.log.LogManager)
[2015-05-30 19:18:55,693] INFO Recovering unflushed segment 0 in log kafkatopic-0. (kafka.log.Log)
[2015-05-30 19:19:05,996] INFO Completed load of log kafkatopic-0 with log end offset 1938914 (kafka.log.Log)
[2015-05-30 19:19:06,009] INFO Recovering unflushed segment 0 in log my-repilcated-topic-0. (kafka.log.Log)
[2015-05-30 19:19:06,016] WARN Error when freeing index buffer (kafka.log.OffsetIndex)
java.lang.NullPointerException
	at kafka.log.OffsetIndex.kafka$log$OffsetIndex$$forceUnmap(OffsetIndex.scala:301)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:283)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:276)
	at kafka.utils.Utils$.inLock(Utils.scala:535)
	at kafka.log.OffsetIndex.resize(OffsetIndex.scala:276)
	at kafka.log.Log.loadSegments(Log.scala:179)
	at kafka.log.Log.<init>(Log.scala:67)
	at kafka.log.LogManager$$anonfun$loadLogs$2$$anonfun$3$$anonfun$apply$7$$anonfun$apply$1.apply$mcV$sp(LogManager.scala:142)
	at kafka.utils.Utils$$anon$1.run(Utils.scala:54)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:19:06,021] INFO Completed load of log my-repilcated-topic-0 with log end offset 19 (kafka.log.Log)
[2015-05-30 19:19:06,031] INFO Recovering unflushed segment 0 in log test-0. (kafka.log.Log)
[2015-05-30 19:19:06,038] WARN Error when freeing index buffer (kafka.log.OffsetIndex)
java.lang.NullPointerException
	at kafka.log.OffsetIndex.kafka$log$OffsetIndex$$forceUnmap(OffsetIndex.scala:301)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:283)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:276)
	at kafka.utils.Utils$.inLock(Utils.scala:535)
	at kafka.log.OffsetIndex.resize(OffsetIndex.scala:276)
	at kafka.log.Log.loadSegments(Log.scala:179)
	at kafka.log.Log.<init>(Log.scala:67)
	at kafka.log.LogManager$$anonfun$loadLogs$2$$anonfun$3$$anonfun$apply$7$$anonfun$apply$1.apply$mcV$sp(LogManager.scala:142)
	at kafka.utils.Utils$$anon$1.run(Utils.scala:54)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:19:06,043] INFO Completed load of log test-0 with log end offset 137 (kafka.log.Log)
[2015-05-30 19:19:06,046] INFO Logs loading complete. (kafka.log.LogManager)
[2015-05-30 19:19:06,047] INFO Starting log cleanup with a period of 300000 ms. (kafka.log.LogManager)
[2015-05-30 19:19:06,050] INFO Starting log flusher with a default period of 9223372036854775807 ms. (kafka.log.LogManager)
[2015-05-30 19:19:06,080] INFO Awaiting socket connections on 0.0.0.0:9092. (kafka.network.Acceptor)
[2015-05-30 19:19:06,081] INFO [Socket Server on Broker 0], Started (kafka.network.SocketServer)
[2015-05-30 19:19:06,143] INFO Will not load MX4J, mx4j-tools.jar is not in the classpath (kafka.utils.Mx4jLoader$)
[2015-05-30 19:19:06,166] INFO 0 successfully elected as leader (kafka.server.ZookeeperLeaderElector)
[2015-05-30 19:19:06,337] INFO New leader is 0 (kafka.server.ZookeeperLeaderElector$LeaderChangeListener)
[2015-05-30 19:19:06,350] INFO Registered broker 0 at path /brokers/ids/0 with address MININT-4IL6B3T.fareast.corp.microsoft.com:9092. (kafka.utils.ZkUtils$)
[2015-05-30 19:19:06,396] INFO [Kafka Server 0], started (kafka.server.KafkaServer)
[2015-05-30 19:19:06,672] INFO [ReplicaFetcherManager on broker 0] Removed fetcher for partitions [my-repilcated-topic,0],[test,0],[kafkatopic,0] (kafka.server.ReplicaFetcherManager)
[2015-05-30 19:19:06,711] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 19:19:06,731] INFO [ReplicaFetcherManager on broker 0] Removed fetcher for partitions [my-repilcated-topic,0],[test,0],[kafkatopic,0] (kafka.server.ReplicaFetcherManager)
[2015-05-30 19:19:06,769] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 19:19:06,883] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:19:35,724] INFO Verifying properties (kafka.utils.VerifiableProperties)
[2015-05-30 19:19:35,752] INFO Property broker.id is overridden to 0 (kafka.utils.VerifiableProperties)
[2015-05-30 19:19:35,753] INFO Property log.cleaner.enable is overridden to false (kafka.utils.VerifiableProperties)
[2015-05-30 19:19:35,753] INFO Property log.dirs is overridden to /tmp/kafka-logs (kafka.utils.VerifiableProperties)
[2015-05-30 19:19:35,753] INFO Property log.retention.check.interval.ms is overridden to 300000 (kafka.utils.VerifiableProperties)
[2015-05-30 19:19:35,754] INFO Property log.retention.hours is overridden to 168 (kafka.utils.VerifiableProperties)
[2015-05-30 19:19:35,754] INFO Property log.segment.bytes is overridden to 1073741824 (kafka.utils.VerifiableProperties)
[2015-05-30 19:19:35,754] INFO Property num.io.threads is overridden to 8 (kafka.utils.VerifiableProperties)
[2015-05-30 19:19:35,754] INFO Property num.network.threads is overridden to 3 (kafka.utils.VerifiableProperties)
[2015-05-30 19:19:35,755] INFO Property num.partitions is overridden to 1 (kafka.utils.VerifiableProperties)
[2015-05-30 19:19:35,755] INFO Property num.recovery.threads.per.data.dir is overridden to 1 (kafka.utils.VerifiableProperties)
[2015-05-30 19:19:35,755] INFO Property port is overridden to 9092 (kafka.utils.VerifiableProperties)
[2015-05-30 19:19:35,756] INFO Property socket.receive.buffer.bytes is overridden to 102400 (kafka.utils.VerifiableProperties)
[2015-05-30 19:19:35,756] INFO Property socket.request.max.bytes is overridden to 104857600 (kafka.utils.VerifiableProperties)
[2015-05-30 19:19:35,757] INFO Property socket.send.buffer.bytes is overridden to 102400 (kafka.utils.VerifiableProperties)
[2015-05-30 19:19:35,757] INFO Property zookeeper.connect is overridden to localhost:2181 (kafka.utils.VerifiableProperties)
[2015-05-30 19:19:35,758] INFO Property zookeeper.connection.timeout.ms is overridden to 6000 (kafka.utils.VerifiableProperties)
[2015-05-30 19:19:35,795] INFO [Kafka Server 0], starting (kafka.server.KafkaServer)
[2015-05-30 19:19:35,797] INFO [Kafka Server 0], Connecting to zookeeper on localhost:2181 (kafka.server.KafkaServer)
[2015-05-30 19:19:36,071] INFO Loading logs. (kafka.log.LogManager)
[2015-05-30 19:19:36,100] INFO Recovering unflushed segment 0 in log kafkatopic-0. (kafka.log.Log)
[2015-05-30 19:19:45,897] INFO Completed load of log kafkatopic-0 with log end offset 1953778 (kafka.log.Log)
[2015-05-30 19:19:45,908] INFO Recovering unflushed segment 0 in log my-repilcated-topic-0. (kafka.log.Log)
[2015-05-30 19:19:45,915] WARN Error when freeing index buffer (kafka.log.OffsetIndex)
java.lang.NullPointerException
	at kafka.log.OffsetIndex.kafka$log$OffsetIndex$$forceUnmap(OffsetIndex.scala:301)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:283)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:276)
	at kafka.utils.Utils$.inLock(Utils.scala:535)
	at kafka.log.OffsetIndex.resize(OffsetIndex.scala:276)
	at kafka.log.Log.loadSegments(Log.scala:179)
	at kafka.log.Log.<init>(Log.scala:67)
	at kafka.log.LogManager$$anonfun$loadLogs$2$$anonfun$3$$anonfun$apply$7$$anonfun$apply$1.apply$mcV$sp(LogManager.scala:142)
	at kafka.utils.Utils$$anon$1.run(Utils.scala:54)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:19:45,922] INFO Completed load of log my-repilcated-topic-0 with log end offset 19 (kafka.log.Log)
[2015-05-30 19:19:45,932] INFO Recovering unflushed segment 0 in log test-0. (kafka.log.Log)
[2015-05-30 19:19:45,940] WARN Error when freeing index buffer (kafka.log.OffsetIndex)
java.lang.NullPointerException
	at kafka.log.OffsetIndex.kafka$log$OffsetIndex$$forceUnmap(OffsetIndex.scala:301)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:283)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:276)
	at kafka.utils.Utils$.inLock(Utils.scala:535)
	at kafka.log.OffsetIndex.resize(OffsetIndex.scala:276)
	at kafka.log.Log.loadSegments(Log.scala:179)
	at kafka.log.Log.<init>(Log.scala:67)
	at kafka.log.LogManager$$anonfun$loadLogs$2$$anonfun$3$$anonfun$apply$7$$anonfun$apply$1.apply$mcV$sp(LogManager.scala:142)
	at kafka.utils.Utils$$anon$1.run(Utils.scala:54)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:19:45,944] INFO Completed load of log test-0 with log end offset 137 (kafka.log.Log)
[2015-05-30 19:19:45,947] INFO Logs loading complete. (kafka.log.LogManager)
[2015-05-30 19:19:45,948] INFO Starting log cleanup with a period of 300000 ms. (kafka.log.LogManager)
[2015-05-30 19:19:45,952] INFO Starting log flusher with a default period of 9223372036854775807 ms. (kafka.log.LogManager)
[2015-05-30 19:19:45,979] INFO Awaiting socket connections on 0.0.0.0:9092. (kafka.network.Acceptor)
[2015-05-30 19:19:45,980] INFO [Socket Server on Broker 0], Started (kafka.network.SocketServer)
[2015-05-30 19:19:46,044] INFO Will not load MX4J, mx4j-tools.jar is not in the classpath (kafka.utils.Mx4jLoader$)
[2015-05-30 19:19:46,068] INFO 0 successfully elected as leader (kafka.server.ZookeeperLeaderElector)
[2015-05-30 19:19:46,233] INFO New leader is 0 (kafka.server.ZookeeperLeaderElector$LeaderChangeListener)
[2015-05-30 19:19:46,245] INFO Registered broker 0 at path /brokers/ids/0 with address MININT-4IL6B3T.fareast.corp.microsoft.com:9092. (kafka.utils.ZkUtils$)
[2015-05-30 19:19:46,295] INFO [Kafka Server 0], started (kafka.server.KafkaServer)
[2015-05-30 19:19:46,586] INFO [ReplicaFetcherManager on broker 0] Removed fetcher for partitions [my-repilcated-topic,0],[test,0],[kafkatopic,0] (kafka.server.ReplicaFetcherManager)
[2015-05-30 19:19:46,604] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:19:46,649] INFO [ReplicaFetcherManager on broker 0] Removed fetcher for partitions [my-repilcated-topic,0],[test,0],[kafkatopic,0] (kafka.server.ReplicaFetcherManager)
[2015-05-30 19:23:04,740] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:23:10,236] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at kafka.utils.Utils$.read(Utils.scala:380)
	at kafka.network.BoundedByteBufferReceive.readFrom(BoundedByteBufferReceive.scala:54)
	at kafka.network.Processor.read(SocketServer.scala:444)
	at kafka.network.Processor.run(SocketServer.scala:340)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:24:07,683] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:24:14,107] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at kafka.utils.Utils$.read(Utils.scala:380)
	at kafka.network.BoundedByteBufferReceive.readFrom(BoundedByteBufferReceive.scala:54)
	at kafka.network.Processor.read(SocketServer.scala:444)
	at kafka.network.Processor.run(SocketServer.scala:340)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:24:24,143] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:24:25,555] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at kafka.utils.Utils$.read(Utils.scala:380)
	at kafka.network.BoundedByteBufferReceive.readFrom(BoundedByteBufferReceive.scala:54)
	at kafka.network.Processor.read(SocketServer.scala:444)
	at kafka.network.Processor.run(SocketServer.scala:340)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:24:56,826] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:25:02,506] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at kafka.utils.Utils$.read(Utils.scala:380)
	at kafka.network.BoundedByteBufferReceive.readFrom(BoundedByteBufferReceive.scala:54)
	at kafka.network.Processor.read(SocketServer.scala:444)
	at kafka.network.Processor.run(SocketServer.scala:340)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:26:06,082] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:26:09,133] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at kafka.utils.Utils$.read(Utils.scala:380)
	at kafka.network.BoundedByteBufferReceive.readFrom(BoundedByteBufferReceive.scala:54)
	at kafka.network.Processor.read(SocketServer.scala:444)
	at kafka.network.Processor.run(SocketServer.scala:340)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:26:16,388] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:26:19,729] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at kafka.utils.Utils$.read(Utils.scala:380)
	at kafka.network.BoundedByteBufferReceive.readFrom(BoundedByteBufferReceive.scala:54)
	at kafka.network.Processor.read(SocketServer.scala:444)
	at kafka.network.Processor.run(SocketServer.scala:340)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:26:27,879] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:26:34,401] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at kafka.utils.Utils$.read(Utils.scala:380)
	at kafka.network.BoundedByteBufferReceive.readFrom(BoundedByteBufferReceive.scala:54)
	at kafka.network.Processor.read(SocketServer.scala:444)
	at kafka.network.Processor.run(SocketServer.scala:340)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:30:27,100] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:30:38,408] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:229)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:31:40,282] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:31:41,790] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:229)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:34:06,381] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:34:16,381] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:229)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:35:04,610] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:35:26,499] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 19:35:55,230] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:229)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:36:43,717] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:37:04,233] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An established connection was aborted by the software in your host machine
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.TopicDataSend.writeTo(FetchResponse.scala:123)
	at kafka.network.MultiSend.writeTo(Transmission.scala:101)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:231)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:37:04,239] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:37:15,076] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:37:15,131] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An established connection was aborted by the software in your host machine
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.TopicDataSend.writeTo(FetchResponse.scala:123)
	at kafka.network.MultiSend.writeTo(Transmission.scala:101)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:231)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:37:26,153] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An established connection was aborted by the software in your host machine
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.TopicDataSend.writeTo(FetchResponse.scala:123)
	at kafka.network.MultiSend.writeTo(Transmission.scala:101)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:231)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:37:26,191] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:38:01,675] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:229)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:38:09,315] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:38:19,309] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:229)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:38:27,172] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:38:53,438] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:38:53,477] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An established connection was aborted by the software in your host machine
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.TopicDataSend.writeTo(FetchResponse.scala:123)
	at kafka.network.MultiSend.writeTo(Transmission.scala:101)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:231)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:39:50,368] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:229)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:39:57,207] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:40:08,736] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:40:08,764] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An established connection was aborted by the software in your host machine
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.TopicDataSend.writeTo(FetchResponse.scala:123)
	at kafka.network.MultiSend.writeTo(Transmission.scala:101)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:231)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:40:28,566] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:229)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:40:36,243] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:40:37,091] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:40:37,134] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An established connection was aborted by the software in your host machine
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.TopicDataSend.writeTo(FetchResponse.scala:123)
	at kafka.network.MultiSend.writeTo(Transmission.scala:101)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:231)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:40:39,812] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 19:40:39,812] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:40:56,038] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:229)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:41:03,177] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:41:10,317] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:41:10,324] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An established connection was aborted by the software in your host machine
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.TopicDataSend.writeTo(FetchResponse.scala:123)
	at kafka.network.MultiSend.writeTo(Transmission.scala:101)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:231)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:41:18,058] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An established connection was aborted by the software in your host machine
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.TopicDataSend.writeTo(FetchResponse.scala:123)
	at kafka.network.MultiSend.writeTo(Transmission.scala:101)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:231)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:41:18,074] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:42:19,864] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An established connection was aborted by the software in your host machine
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.TopicDataSend.writeTo(FetchResponse.scala:123)
	at kafka.network.MultiSend.writeTo(Transmission.scala:101)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:231)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:42:19,883] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:42:36,342] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:229)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:43:01,054] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An established connection was aborted by the software in your host machine
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.TopicDataSend.writeTo(FetchResponse.scala:123)
	at kafka.network.MultiSend.writeTo(Transmission.scala:101)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:231)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:43:01,060] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:43:03,163] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An established connection was aborted by the software in your host machine
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.TopicDataSend.writeTo(FetchResponse.scala:123)
	at kafka.network.MultiSend.writeTo(Transmission.scala:101)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:231)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:43:03,195] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:43:21,400] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An established connection was aborted by the software in your host machine
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.TopicDataSend.writeTo(FetchResponse.scala:123)
	at kafka.network.MultiSend.writeTo(Transmission.scala:101)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:231)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:43:21,445] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:43:24,135] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:43:24,150] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An established connection was aborted by the software in your host machine
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.TopicDataSend.writeTo(FetchResponse.scala:123)
	at kafka.network.MultiSend.writeTo(Transmission.scala:101)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:231)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:43:42,088] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An established connection was aborted by the software in your host machine
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.TopicDataSend.writeTo(FetchResponse.scala:123)
	at kafka.network.MultiSend.writeTo(Transmission.scala:101)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:231)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:43:42,093] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:43:46,292] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:43:46,298] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An established connection was aborted by the software in your host machine
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.TopicDataSend.writeTo(FetchResponse.scala:123)
	at kafka.network.MultiSend.writeTo(Transmission.scala:101)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:231)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:44:24,083] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:44:24,085] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An established connection was aborted by the software in your host machine
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.TopicDataSend.writeTo(FetchResponse.scala:123)
	at kafka.network.MultiSend.writeTo(Transmission.scala:101)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:231)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:44:44,489] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An established connection was aborted by the software in your host machine
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.TopicDataSend.writeTo(FetchResponse.scala:123)
	at kafka.network.MultiSend.writeTo(Transmission.scala:101)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:231)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:44:44,494] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:45:17,083] INFO Verifying properties (kafka.utils.VerifiableProperties)
[2015-05-30 19:45:17,114] INFO Property broker.id is overridden to 0 (kafka.utils.VerifiableProperties)
[2015-05-30 19:45:17,114] INFO Property log.cleaner.enable is overridden to false (kafka.utils.VerifiableProperties)
[2015-05-30 19:45:17,114] INFO Property log.dirs is overridden to /tmp/kafka-logs (kafka.utils.VerifiableProperties)
[2015-05-30 19:45:17,115] INFO Property log.retention.check.interval.ms is overridden to 300000 (kafka.utils.VerifiableProperties)
[2015-05-30 19:45:17,115] INFO Property log.retention.hours is overridden to 168 (kafka.utils.VerifiableProperties)
[2015-05-30 19:45:17,115] INFO Property log.segment.bytes is overridden to 1073741824 (kafka.utils.VerifiableProperties)
[2015-05-30 19:45:17,115] INFO Property num.io.threads is overridden to 8 (kafka.utils.VerifiableProperties)
[2015-05-30 19:45:17,116] INFO Property num.network.threads is overridden to 3 (kafka.utils.VerifiableProperties)
[2015-05-30 19:45:17,116] INFO Property num.partitions is overridden to 1 (kafka.utils.VerifiableProperties)
[2015-05-30 19:45:17,116] INFO Property num.recovery.threads.per.data.dir is overridden to 1 (kafka.utils.VerifiableProperties)
[2015-05-30 19:45:17,116] INFO Property port is overridden to 9092 (kafka.utils.VerifiableProperties)
[2015-05-30 19:45:17,117] INFO Property socket.receive.buffer.bytes is overridden to 102400 (kafka.utils.VerifiableProperties)
[2015-05-30 19:45:17,117] INFO Property socket.request.max.bytes is overridden to 104857600 (kafka.utils.VerifiableProperties)
[2015-05-30 19:45:17,118] INFO Property socket.send.buffer.bytes is overridden to 102400 (kafka.utils.VerifiableProperties)
[2015-05-30 19:45:17,118] INFO Property zookeeper.connect is overridden to localhost:2181 (kafka.utils.VerifiableProperties)
[2015-05-30 19:45:17,119] INFO Property zookeeper.connection.timeout.ms is overridden to 6000 (kafka.utils.VerifiableProperties)
[2015-05-30 19:45:17,155] INFO [Kafka Server 0], starting (kafka.server.KafkaServer)
[2015-05-30 19:45:17,158] INFO [Kafka Server 0], Connecting to zookeeper on localhost:2181 (kafka.server.KafkaServer)
[2015-05-30 19:45:17,418] INFO Loading logs. (kafka.log.LogManager)
[2015-05-30 19:45:17,448] INFO Recovering unflushed segment 0 in log kafkatopic-0. (kafka.log.Log)
[2015-05-30 19:45:27,510] INFO Completed load of log kafkatopic-0 with log end offset 1953998 (kafka.log.Log)
[2015-05-30 19:45:27,519] INFO Recovering unflushed segment 0 in log my-repilcated-topic-0. (kafka.log.Log)
[2015-05-30 19:45:27,525] WARN Error when freeing index buffer (kafka.log.OffsetIndex)
java.lang.NullPointerException
	at kafka.log.OffsetIndex.kafka$log$OffsetIndex$$forceUnmap(OffsetIndex.scala:301)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:283)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:276)
	at kafka.utils.Utils$.inLock(Utils.scala:535)
	at kafka.log.OffsetIndex.resize(OffsetIndex.scala:276)
	at kafka.log.Log.loadSegments(Log.scala:179)
	at kafka.log.Log.<init>(Log.scala:67)
	at kafka.log.LogManager$$anonfun$loadLogs$2$$anonfun$3$$anonfun$apply$7$$anonfun$apply$1.apply$mcV$sp(LogManager.scala:142)
	at kafka.utils.Utils$$anon$1.run(Utils.scala:54)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:45:27,529] INFO Completed load of log my-repilcated-topic-0 with log end offset 19 (kafka.log.Log)
[2015-05-30 19:45:27,537] INFO Recovering unflushed segment 0 in log test-0. (kafka.log.Log)
[2015-05-30 19:45:27,543] WARN Error when freeing index buffer (kafka.log.OffsetIndex)
java.lang.NullPointerException
	at kafka.log.OffsetIndex.kafka$log$OffsetIndex$$forceUnmap(OffsetIndex.scala:301)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:283)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:276)
	at kafka.utils.Utils$.inLock(Utils.scala:535)
	at kafka.log.OffsetIndex.resize(OffsetIndex.scala:276)
	at kafka.log.Log.loadSegments(Log.scala:179)
	at kafka.log.Log.<init>(Log.scala:67)
	at kafka.log.LogManager$$anonfun$loadLogs$2$$anonfun$3$$anonfun$apply$7$$anonfun$apply$1.apply$mcV$sp(LogManager.scala:142)
	at kafka.utils.Utils$$anon$1.run(Utils.scala:54)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:45:27,546] INFO Completed load of log test-0 with log end offset 137 (kafka.log.Log)
[2015-05-30 19:45:27,550] INFO Logs loading complete. (kafka.log.LogManager)
[2015-05-30 19:45:27,550] INFO Starting log cleanup with a period of 300000 ms. (kafka.log.LogManager)
[2015-05-30 19:45:27,552] INFO Starting log flusher with a default period of 9223372036854775807 ms. (kafka.log.LogManager)
[2015-05-30 19:45:27,578] INFO Awaiting socket connections on 0.0.0.0:9092. (kafka.network.Acceptor)
[2015-05-30 19:45:27,579] INFO [Socket Server on Broker 0], Started (kafka.network.SocketServer)
[2015-05-30 19:45:27,627] INFO Will not load MX4J, mx4j-tools.jar is not in the classpath (kafka.utils.Mx4jLoader$)
[2015-05-30 19:45:27,650] INFO 0 successfully elected as leader (kafka.server.ZookeeperLeaderElector)
[2015-05-30 19:45:27,839] ERROR [KafkaApi-0] error when handling request Name: TopicMetadataRequest; Version: 0; CorrelationId: 2; ClientId: console-producer; Topics: kafkatopic (kafka.server.KafkaApis)
kafka.admin.AdminOperationException: replication factor: 1 larger than available brokers: 0
	at kafka.admin.AdminUtils$.assignReplicasToBrokers(AdminUtils.scala:70)
	at kafka.admin.AdminUtils$.createTopic(AdminUtils.scala:171)
	at kafka.server.KafkaApis$$anonfun$19.apply(KafkaApis.scala:520)
	at kafka.server.KafkaApis$$anonfun$19.apply(KafkaApis.scala:503)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
	at scala.collection.immutable.Set$Set1.foreach(Set.scala:74)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:244)
	at scala.collection.AbstractSet.scala$collection$SetLike$$super$map(Set.scala:47)
	at scala.collection.SetLike$class.map(SetLike.scala:93)
	at scala.collection.AbstractSet.map(Set.scala:47)
	at kafka.server.KafkaApis.getTopicMetadata(KafkaApis.scala:503)
	at kafka.server.KafkaApis.handleTopicMetadataRequest(KafkaApis.scala:542)
	at kafka.server.KafkaApis.handle(KafkaApis.scala:62)
	at kafka.server.KafkaRequestHandler.run(KafkaRequestHandler.scala:59)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:45:27,905] ERROR [KafkaApi-0] error when handling request Name: TopicMetadataRequest; Version: 0; CorrelationId: 3; ClientId: console-producer; Topics: kafkatopic (kafka.server.KafkaApis)
kafka.admin.AdminOperationException: replication factor: 1 larger than available brokers: 0
	at kafka.admin.AdminUtils$.assignReplicasToBrokers(AdminUtils.scala:70)
	at kafka.admin.AdminUtils$.createTopic(AdminUtils.scala:171)
	at kafka.server.KafkaApis$$anonfun$19.apply(KafkaApis.scala:520)
	at kafka.server.KafkaApis$$anonfun$19.apply(KafkaApis.scala:503)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
	at scala.collection.immutable.Set$Set1.foreach(Set.scala:74)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:244)
	at scala.collection.AbstractSet.scala$collection$SetLike$$super$map(Set.scala:47)
	at scala.collection.SetLike$class.map(SetLike.scala:93)
	at scala.collection.AbstractSet.map(Set.scala:47)
	at kafka.server.KafkaApis.getTopicMetadata(KafkaApis.scala:503)
	at kafka.server.KafkaApis.handleTopicMetadataRequest(KafkaApis.scala:542)
	at kafka.server.KafkaApis.handle(KafkaApis.scala:62)
	at kafka.server.KafkaRequestHandler.run(KafkaRequestHandler.scala:59)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:45:27,920] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 19:45:27,927] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 19:45:27,933] INFO New leader is 0 (kafka.server.ZookeeperLeaderElector$LeaderChangeListener)
[2015-05-30 19:45:27,944] INFO Registered broker 0 at path /brokers/ids/0 with address MININT-4IL6B3T.fareast.corp.microsoft.com:9092. (kafka.utils.ZkUtils$)
[2015-05-30 19:45:27,955] INFO [Kafka Server 0], started (kafka.server.KafkaServer)
[2015-05-30 19:45:28,060] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 19:45:28,070] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:45:28,079] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 19:45:28,153] INFO [ReplicaFetcherManager on broker 0] Removed fetcher for partitions [my-repilcated-topic,0],[test,0],[kafkatopic,0] (kafka.server.ReplicaFetcherManager)
[2015-05-30 19:45:28,197] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 19:45:28,240] INFO [ReplicaFetcherManager on broker 0] Removed fetcher for partitions [my-repilcated-topic,0],[test,0],[kafkatopic,0] (kafka.server.ReplicaFetcherManager)
[2015-05-30 19:45:28,277] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:45:28,302] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:45:30,854] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at kafka.utils.Utils$.read(Utils.scala:380)
	at kafka.network.BoundedByteBufferReceive.readFrom(BoundedByteBufferReceive.scala:54)
	at kafka.network.Processor.read(SocketServer.scala:444)
	at kafka.network.Processor.run(SocketServer.scala:340)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:45:31,957] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:229)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:45:35,974] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 19:45:39,755] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:45:43,621] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:229)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:45:48,707] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:45:48,727] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An established connection was aborted by the software in your host machine
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.TopicDataSend.writeTo(FetchResponse.scala:123)
	at kafka.network.MultiSend.writeTo(Transmission.scala:101)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:231)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:46:27,060] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:46:27,102] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An established connection was aborted by the software in your host machine
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.TopicDataSend.writeTo(FetchResponse.scala:123)
	at kafka.network.MultiSend.writeTo(Transmission.scala:101)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:231)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:46:39,879] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An established connection was aborted by the software in your host machine
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.TopicDataSend.writeTo(FetchResponse.scala:123)
	at kafka.network.MultiSend.writeTo(Transmission.scala:101)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:231)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:46:39,911] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:47:18,443] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:47:18,445] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An established connection was aborted by the software in your host machine
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.TopicDataSend.writeTo(FetchResponse.scala:123)
	at kafka.network.MultiSend.writeTo(Transmission.scala:101)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:231)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:49:26,693] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:229)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:49:31,068] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at kafka.utils.Utils$.read(Utils.scala:380)
	at kafka.network.BoundedByteBufferReceive.readFrom(BoundedByteBufferReceive.scala:54)
	at kafka.network.Processor.read(SocketServer.scala:444)
	at kafka.network.Processor.run(SocketServer.scala:340)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:50:26,142] INFO Verifying properties (kafka.utils.VerifiableProperties)
[2015-05-30 19:50:26,158] INFO Property broker.id is overridden to 0 (kafka.utils.VerifiableProperties)
[2015-05-30 19:50:26,158] INFO Property log.cleaner.enable is overridden to false (kafka.utils.VerifiableProperties)
[2015-05-30 19:50:26,158] INFO Property log.dirs is overridden to /tmp/kafka-logs (kafka.utils.VerifiableProperties)
[2015-05-30 19:50:26,158] INFO Property log.retention.check.interval.ms is overridden to 300000 (kafka.utils.VerifiableProperties)
[2015-05-30 19:50:26,158] INFO Property log.retention.hours is overridden to 168 (kafka.utils.VerifiableProperties)
[2015-05-30 19:50:26,158] INFO Property log.segment.bytes is overridden to 1073741824 (kafka.utils.VerifiableProperties)
[2015-05-30 19:50:26,158] INFO Property num.io.threads is overridden to 8 (kafka.utils.VerifiableProperties)
[2015-05-30 19:50:26,158] INFO Property num.network.threads is overridden to 3 (kafka.utils.VerifiableProperties)
[2015-05-30 19:50:26,158] INFO Property num.partitions is overridden to 1 (kafka.utils.VerifiableProperties)
[2015-05-30 19:50:26,158] INFO Property num.recovery.threads.per.data.dir is overridden to 1 (kafka.utils.VerifiableProperties)
[2015-05-30 19:50:26,158] INFO Property port is overridden to 9092 (kafka.utils.VerifiableProperties)
[2015-05-30 19:50:26,158] INFO Property socket.receive.buffer.bytes is overridden to 102400 (kafka.utils.VerifiableProperties)
[2015-05-30 19:50:26,158] INFO Property socket.request.max.bytes is overridden to 104857600 (kafka.utils.VerifiableProperties)
[2015-05-30 19:50:26,158] INFO Property socket.send.buffer.bytes is overridden to 102400 (kafka.utils.VerifiableProperties)
[2015-05-30 19:50:26,158] INFO Property zookeeper.connect is overridden to localhost:2181 (kafka.utils.VerifiableProperties)
[2015-05-30 19:50:26,174] INFO Property zookeeper.connection.timeout.ms is overridden to 6000 (kafka.utils.VerifiableProperties)
[2015-05-30 19:50:26,205] INFO [Kafka Server 0], starting (kafka.server.KafkaServer)
[2015-05-30 19:50:26,205] INFO [Kafka Server 0], Connecting to zookeeper on localhost:2181 (kafka.server.KafkaServer)
[2015-05-30 19:50:26,549] INFO Loading logs. (kafka.log.LogManager)
[2015-05-30 19:50:26,580] INFO Recovering unflushed segment 0 in log kafkatopic-0. (kafka.log.Log)
[2015-05-30 19:50:38,389] INFO Completed load of log kafkatopic-0 with log end offset 1954077 (kafka.log.Log)
[2015-05-30 19:50:38,405] INFO Recovering unflushed segment 0 in log my-repilcated-topic-0. (kafka.log.Log)
[2015-05-30 19:50:38,405] WARN Error when freeing index buffer (kafka.log.OffsetIndex)
java.lang.NullPointerException
	at kafka.log.OffsetIndex.kafka$log$OffsetIndex$$forceUnmap(OffsetIndex.scala:301)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:283)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:276)
	at kafka.utils.Utils$.inLock(Utils.scala:535)
	at kafka.log.OffsetIndex.resize(OffsetIndex.scala:276)
	at kafka.log.Log.loadSegments(Log.scala:179)
	at kafka.log.Log.<init>(Log.scala:67)
	at kafka.log.LogManager$$anonfun$loadLogs$2$$anonfun$3$$anonfun$apply$7$$anonfun$apply$1.apply$mcV$sp(LogManager.scala:142)
	at kafka.utils.Utils$$anon$1.run(Utils.scala:54)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:50:38,420] INFO Completed load of log my-repilcated-topic-0 with log end offset 19 (kafka.log.Log)
[2015-05-30 19:50:38,420] INFO Recovering unflushed segment 0 in log test-0. (kafka.log.Log)
[2015-05-30 19:50:38,436] WARN Error when freeing index buffer (kafka.log.OffsetIndex)
java.lang.NullPointerException
	at kafka.log.OffsetIndex.kafka$log$OffsetIndex$$forceUnmap(OffsetIndex.scala:301)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:283)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:276)
	at kafka.utils.Utils$.inLock(Utils.scala:535)
	at kafka.log.OffsetIndex.resize(OffsetIndex.scala:276)
	at kafka.log.Log.loadSegments(Log.scala:179)
	at kafka.log.Log.<init>(Log.scala:67)
	at kafka.log.LogManager$$anonfun$loadLogs$2$$anonfun$3$$anonfun$apply$7$$anonfun$apply$1.apply$mcV$sp(LogManager.scala:142)
	at kafka.utils.Utils$$anon$1.run(Utils.scala:54)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:50:38,436] INFO Completed load of log test-0 with log end offset 137 (kafka.log.Log)
[2015-05-30 19:50:38,436] INFO Logs loading complete. (kafka.log.LogManager)
[2015-05-30 19:50:38,436] INFO Starting log cleanup with a period of 300000 ms. (kafka.log.LogManager)
[2015-05-30 19:50:38,452] INFO Starting log flusher with a default period of 9223372036854775807 ms. (kafka.log.LogManager)
[2015-05-30 19:50:38,483] INFO Awaiting socket connections on 0.0.0.0:9092. (kafka.network.Acceptor)
[2015-05-30 19:50:38,483] INFO [Socket Server on Broker 0], Started (kafka.network.SocketServer)
[2015-05-30 19:50:38,530] INFO Will not load MX4J, mx4j-tools.jar is not in the classpath (kafka.utils.Mx4jLoader$)
[2015-05-30 19:50:38,561] INFO 0 successfully elected as leader (kafka.server.ZookeeperLeaderElector)
[2015-05-30 19:50:38,733] INFO New leader is 0 (kafka.server.ZookeeperLeaderElector$LeaderChangeListener)
[2015-05-30 19:50:38,749] INFO Registered broker 0 at path /brokers/ids/0 with address MININT-4IL6B3T.fareast.corp.microsoft.com:9092. (kafka.utils.ZkUtils$)
[2015-05-30 19:50:38,858] INFO [Kafka Server 0], started (kafka.server.KafkaServer)
[2015-05-30 19:50:39,106] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 19:50:39,126] INFO [ReplicaFetcherManager on broker 0] Removed fetcher for partitions [my-repilcated-topic,0],[test,0],[kafkatopic,0] (kafka.server.ReplicaFetcherManager)
[2015-05-30 19:50:39,135] WARN [KafkaApi-0] Produce request with correlation id 9 from client console-producer on partition [kafkatopic,0] failed due to Leader not local for partition [kafkatopic,0] on broker 0 (kafka.server.KafkaApis)
[2015-05-30 19:50:39,142] INFO [KafkaApi-0] Send the close connection response due to error handling produce request [clientId = console-producer, correlationId = 9, topicAndPartition = [kafkatopic,0]] with Ack=0 (kafka.server.KafkaApis)
[2015-05-30 19:50:39,161] INFO [ReplicaFetcherManager on broker 0] Removed fetcher for partitions [my-repilcated-topic,0],[test,0],[kafkatopic,0] (kafka.server.ReplicaFetcherManager)
[2015-05-30 19:50:39,170] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:50:51,542] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:229)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:51:01,156] INFO Verifying properties (kafka.utils.VerifiableProperties)
[2015-05-30 19:51:01,187] INFO Property broker.id is overridden to 0 (kafka.utils.VerifiableProperties)
[2015-05-30 19:51:01,187] INFO Property log.cleaner.enable is overridden to false (kafka.utils.VerifiableProperties)
[2015-05-30 19:51:01,187] INFO Property log.dirs is overridden to /tmp/kafka-logs (kafka.utils.VerifiableProperties)
[2015-05-30 19:51:01,187] INFO Property log.retention.check.interval.ms is overridden to 300000 (kafka.utils.VerifiableProperties)
[2015-05-30 19:51:01,187] INFO Property log.retention.hours is overridden to 168 (kafka.utils.VerifiableProperties)
[2015-05-30 19:51:01,187] INFO Property log.segment.bytes is overridden to 1073741824 (kafka.utils.VerifiableProperties)
[2015-05-30 19:51:01,187] INFO Property num.io.threads is overridden to 8 (kafka.utils.VerifiableProperties)
[2015-05-30 19:51:01,187] INFO Property num.network.threads is overridden to 3 (kafka.utils.VerifiableProperties)
[2015-05-30 19:51:01,187] INFO Property num.partitions is overridden to 1 (kafka.utils.VerifiableProperties)
[2015-05-30 19:51:01,187] INFO Property num.recovery.threads.per.data.dir is overridden to 1 (kafka.utils.VerifiableProperties)
[2015-05-30 19:51:01,203] INFO Property port is overridden to 9092 (kafka.utils.VerifiableProperties)
[2015-05-30 19:51:01,203] INFO Property socket.receive.buffer.bytes is overridden to 102400 (kafka.utils.VerifiableProperties)
[2015-05-30 19:51:01,203] INFO Property socket.request.max.bytes is overridden to 104857600 (kafka.utils.VerifiableProperties)
[2015-05-30 19:51:01,203] INFO Property socket.send.buffer.bytes is overridden to 102400 (kafka.utils.VerifiableProperties)
[2015-05-30 19:51:01,203] INFO Property zookeeper.connect is overridden to localhost:2181 (kafka.utils.VerifiableProperties)
[2015-05-30 19:51:01,203] INFO Property zookeeper.connection.timeout.ms is overridden to 6000 (kafka.utils.VerifiableProperties)
[2015-05-30 19:51:01,249] INFO [Kafka Server 0], starting (kafka.server.KafkaServer)
[2015-05-30 19:51:01,249] INFO [Kafka Server 0], Connecting to zookeeper on localhost:2181 (kafka.server.KafkaServer)
[2015-05-30 19:51:01,578] INFO Loading logs. (kafka.log.LogManager)
[2015-05-30 19:51:01,609] INFO Recovering unflushed segment 0 in log kafkatopic-0. (kafka.log.Log)
[2015-05-30 19:51:13,612] INFO Completed load of log kafkatopic-0 with log end offset 1954077 (kafka.log.Log)
[2015-05-30 19:51:13,628] INFO Recovering unflushed segment 0 in log my-repilcated-topic-0. (kafka.log.Log)
[2015-05-30 19:51:13,628] WARN Error when freeing index buffer (kafka.log.OffsetIndex)
java.lang.NullPointerException
	at kafka.log.OffsetIndex.kafka$log$OffsetIndex$$forceUnmap(OffsetIndex.scala:301)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:283)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:276)
	at kafka.utils.Utils$.inLock(Utils.scala:535)
	at kafka.log.OffsetIndex.resize(OffsetIndex.scala:276)
	at kafka.log.Log.loadSegments(Log.scala:179)
	at kafka.log.Log.<init>(Log.scala:67)
	at kafka.log.LogManager$$anonfun$loadLogs$2$$anonfun$3$$anonfun$apply$7$$anonfun$apply$1.apply$mcV$sp(LogManager.scala:142)
	at kafka.utils.Utils$$anon$1.run(Utils.scala:54)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:51:13,644] INFO Completed load of log my-repilcated-topic-0 with log end offset 19 (kafka.log.Log)
[2015-05-30 19:51:13,644] INFO Recovering unflushed segment 0 in log test-0. (kafka.log.Log)
[2015-05-30 19:51:13,659] WARN Error when freeing index buffer (kafka.log.OffsetIndex)
java.lang.NullPointerException
	at kafka.log.OffsetIndex.kafka$log$OffsetIndex$$forceUnmap(OffsetIndex.scala:301)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:283)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:276)
	at kafka.utils.Utils$.inLock(Utils.scala:535)
	at kafka.log.OffsetIndex.resize(OffsetIndex.scala:276)
	at kafka.log.Log.loadSegments(Log.scala:179)
	at kafka.log.Log.<init>(Log.scala:67)
	at kafka.log.LogManager$$anonfun$loadLogs$2$$anonfun$3$$anonfun$apply$7$$anonfun$apply$1.apply$mcV$sp(LogManager.scala:142)
	at kafka.utils.Utils$$anon$1.run(Utils.scala:54)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:51:13,659] INFO Completed load of log test-0 with log end offset 137 (kafka.log.Log)
[2015-05-30 19:51:13,659] INFO Logs loading complete. (kafka.log.LogManager)
[2015-05-30 19:51:13,659] INFO Starting log cleanup with a period of 300000 ms. (kafka.log.LogManager)
[2015-05-30 19:51:13,659] INFO Starting log flusher with a default period of 9223372036854775807 ms. (kafka.log.LogManager)
[2015-05-30 19:51:13,691] INFO Awaiting socket connections on 0.0.0.0:9092. (kafka.network.Acceptor)
[2015-05-30 19:51:13,691] INFO [Socket Server on Broker 0], Started (kafka.network.SocketServer)
[2015-05-30 19:51:13,753] INFO Will not load MX4J, mx4j-tools.jar is not in the classpath (kafka.utils.Mx4jLoader$)
[2015-05-30 19:51:13,784] INFO 0 successfully elected as leader (kafka.server.ZookeeperLeaderElector)
[2015-05-30 19:51:13,831] ERROR [KafkaApi-0] error when handling request Name: TopicMetadataRequest; Version: 0; CorrelationId: 4; ClientId: console-producer; Topics: kafkatopic (kafka.server.KafkaApis)
kafka.admin.AdminOperationException: replication factor: 1 larger than available brokers: 0
	at kafka.admin.AdminUtils$.assignReplicasToBrokers(AdminUtils.scala:70)
	at kafka.admin.AdminUtils$.createTopic(AdminUtils.scala:171)
	at kafka.server.KafkaApis$$anonfun$19.apply(KafkaApis.scala:520)
	at kafka.server.KafkaApis$$anonfun$19.apply(KafkaApis.scala:503)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
	at scala.collection.immutable.Set$Set1.foreach(Set.scala:74)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:244)
	at scala.collection.AbstractSet.scala$collection$SetLike$$super$map(Set.scala:47)
	at scala.collection.SetLike$class.map(SetLike.scala:93)
	at scala.collection.AbstractSet.map(Set.scala:47)
	at kafka.server.KafkaApis.getTopicMetadata(KafkaApis.scala:503)
	at kafka.server.KafkaApis.handleTopicMetadataRequest(KafkaApis.scala:542)
	at kafka.server.KafkaApis.handle(KafkaApis.scala:62)
	at kafka.server.KafkaRequestHandler.run(KafkaRequestHandler.scala:59)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:51:13,862] ERROR [KafkaApi-0] error when handling request Name: TopicMetadataRequest; Version: 0; CorrelationId: 5; ClientId: console-producer; Topics: kafkatopic (kafka.server.KafkaApis)
kafka.admin.AdminOperationException: replication factor: 1 larger than available brokers: 0
	at kafka.admin.AdminUtils$.assignReplicasToBrokers(AdminUtils.scala:70)
	at kafka.admin.AdminUtils$.createTopic(AdminUtils.scala:171)
	at kafka.server.KafkaApis$$anonfun$19.apply(KafkaApis.scala:520)
	at kafka.server.KafkaApis$$anonfun$19.apply(KafkaApis.scala:503)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
	at scala.collection.immutable.Set$Set1.foreach(Set.scala:74)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:244)
	at scala.collection.AbstractSet.scala$collection$SetLike$$super$map(Set.scala:47)
	at scala.collection.SetLike$class.map(SetLike.scala:93)
	at scala.collection.AbstractSet.map(Set.scala:47)
	at kafka.server.KafkaApis.getTopicMetadata(KafkaApis.scala:503)
	at kafka.server.KafkaApis.handleTopicMetadataRequest(KafkaApis.scala:542)
	at kafka.server.KafkaApis.handle(KafkaApis.scala:62)
	at kafka.server.KafkaRequestHandler.run(KafkaRequestHandler.scala:59)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:51:13,894] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 19:51:13,894] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 19:51:13,987] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 19:51:14,003] ERROR [KafkaApi-0] error when handling request Name: TopicMetadataRequest; Version: 0; CorrelationId: 6; ClientId: console-producer; Topics: kafkatopic (kafka.server.KafkaApis)
kafka.admin.AdminOperationException: replication factor: 1 larger than available brokers: 0
	at kafka.admin.AdminUtils$.assignReplicasToBrokers(AdminUtils.scala:70)
	at kafka.admin.AdminUtils$.createTopic(AdminUtils.scala:171)
	at kafka.server.KafkaApis$$anonfun$19.apply(KafkaApis.scala:520)
	at kafka.server.KafkaApis$$anonfun$19.apply(KafkaApis.scala:503)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
	at scala.collection.immutable.Set$Set1.foreach(Set.scala:74)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:244)
	at scala.collection.AbstractSet.scala$collection$SetLike$$super$map(Set.scala:47)
	at scala.collection.SetLike$class.map(SetLike.scala:93)
	at scala.collection.AbstractSet.map(Set.scala:47)
	at kafka.server.KafkaApis.getTopicMetadata(KafkaApis.scala:503)
	at kafka.server.KafkaApis.handleTopicMetadataRequest(KafkaApis.scala:542)
	at kafka.server.KafkaApis.handle(KafkaApis.scala:62)
	at kafka.server.KafkaRequestHandler.run(KafkaRequestHandler.scala:59)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:51:14,003] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 19:51:14,003] ERROR [KafkaApi-0] error when handling request Name: TopicMetadataRequest; Version: 0; CorrelationId: 7; ClientId: console-producer; Topics: kafkatopic (kafka.server.KafkaApis)
kafka.admin.AdminOperationException: replication factor: 1 larger than available brokers: 0
	at kafka.admin.AdminUtils$.assignReplicasToBrokers(AdminUtils.scala:70)
	at kafka.admin.AdminUtils$.createTopic(AdminUtils.scala:171)
	at kafka.server.KafkaApis$$anonfun$19.apply(KafkaApis.scala:520)
	at kafka.server.KafkaApis$$anonfun$19.apply(KafkaApis.scala:503)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
	at scala.collection.immutable.Set$Set1.foreach(Set.scala:74)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:244)
	at scala.collection.AbstractSet.scala$collection$SetLike$$super$map(Set.scala:47)
	at scala.collection.SetLike$class.map(SetLike.scala:93)
	at scala.collection.AbstractSet.map(Set.scala:47)
	at kafka.server.KafkaApis.getTopicMetadata(KafkaApis.scala:503)
	at kafka.server.KafkaApis.handleTopicMetadataRequest(KafkaApis.scala:542)
	at kafka.server.KafkaApis.handle(KafkaApis.scala:62)
	at kafka.server.KafkaRequestHandler.run(KafkaRequestHandler.scala:59)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:51:14,050] INFO New leader is 0 (kafka.server.ZookeeperLeaderElector$LeaderChangeListener)
[2015-05-30 19:51:14,066] INFO Registered broker 0 at path /brokers/ids/0 with address MININT-4IL6B3T.fareast.corp.microsoft.com:9092. (kafka.utils.ZkUtils$)
[2015-05-30 19:51:14,081] INFO [Kafka Server 0], started (kafka.server.KafkaServer)
[2015-05-30 19:51:14,128] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 19:51:14,159] INFO [ReplicaFetcherManager on broker 0] Removed fetcher for partitions [my-repilcated-topic,0],[test,0],[kafkatopic,0] (kafka.server.ReplicaFetcherManager)
[2015-05-30 19:51:14,191] INFO [ReplicaFetcherManager on broker 0] Removed fetcher for partitions [my-repilcated-topic,0],[test,0],[kafkatopic,0] (kafka.server.ReplicaFetcherManager)
[2015-05-30 19:51:15,175] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 19:51:30,395] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at kafka.utils.Utils$.read(Utils.scala:380)
	at kafka.network.BoundedByteBufferReceive.readFrom(BoundedByteBufferReceive.scala:54)
	at kafka.network.Processor.read(SocketServer.scala:444)
	at kafka.network.Processor.run(SocketServer.scala:340)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:51:40,207] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 19:51:45,566] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:54:53,741] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 19:54:57,589] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at kafka.utils.Utils$.read(Utils.scala:380)
	at kafka.network.BoundedByteBufferReceive.readFrom(BoundedByteBufferReceive.scala:54)
	at kafka.network.Processor.read(SocketServer.scala:444)
	at kafka.network.Processor.run(SocketServer.scala:340)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:55:46,791] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:55:50,322] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 19:55:52,831] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at kafka.utils.Utils$.read(Utils.scala:380)
	at kafka.network.BoundedByteBufferReceive.readFrom(BoundedByteBufferReceive.scala:54)
	at kafka.network.Processor.read(SocketServer.scala:444)
	at kafka.network.Processor.run(SocketServer.scala:340)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:56:01,072] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:56:09,880] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at kafka.utils.Utils$.read(Utils.scala:380)
	at kafka.network.BoundedByteBufferReceive.readFrom(BoundedByteBufferReceive.scala:54)
	at kafka.network.Processor.read(SocketServer.scala:444)
	at kafka.network.Processor.run(SocketServer.scala:340)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:56:59,244] INFO Verifying properties (kafka.utils.VerifiableProperties)
[2015-05-30 19:56:59,259] INFO Property broker.id is overridden to 0 (kafka.utils.VerifiableProperties)
[2015-05-30 19:56:59,259] INFO Property log.cleaner.enable is overridden to false (kafka.utils.VerifiableProperties)
[2015-05-30 19:56:59,259] INFO Property log.dirs is overridden to /tmp/kafka-logs (kafka.utils.VerifiableProperties)
[2015-05-30 19:56:59,259] INFO Property log.retention.check.interval.ms is overridden to 300000 (kafka.utils.VerifiableProperties)
[2015-05-30 19:56:59,259] INFO Property log.retention.hours is overridden to 168 (kafka.utils.VerifiableProperties)
[2015-05-30 19:56:59,259] INFO Property log.segment.bytes is overridden to 1073741824 (kafka.utils.VerifiableProperties)
[2015-05-30 19:56:59,259] INFO Property num.io.threads is overridden to 8 (kafka.utils.VerifiableProperties)
[2015-05-30 19:56:59,259] INFO Property num.network.threads is overridden to 3 (kafka.utils.VerifiableProperties)
[2015-05-30 19:56:59,259] INFO Property num.partitions is overridden to 1 (kafka.utils.VerifiableProperties)
[2015-05-30 19:56:59,259] INFO Property num.recovery.threads.per.data.dir is overridden to 1 (kafka.utils.VerifiableProperties)
[2015-05-30 19:56:59,275] INFO Property port is overridden to 9092 (kafka.utils.VerifiableProperties)
[2015-05-30 19:56:59,275] INFO Property socket.receive.buffer.bytes is overridden to 102400 (kafka.utils.VerifiableProperties)
[2015-05-30 19:56:59,275] INFO Property socket.request.max.bytes is overridden to 104857600 (kafka.utils.VerifiableProperties)
[2015-05-30 19:56:59,275] INFO Property socket.send.buffer.bytes is overridden to 102400 (kafka.utils.VerifiableProperties)
[2015-05-30 19:56:59,275] INFO Property zookeeper.connect is overridden to localhost:2181 (kafka.utils.VerifiableProperties)
[2015-05-30 19:56:59,275] INFO Property zookeeper.connection.timeout.ms is overridden to 6000 (kafka.utils.VerifiableProperties)
[2015-05-30 19:56:59,306] INFO [Kafka Server 0], starting (kafka.server.KafkaServer)
[2015-05-30 19:56:59,306] INFO [Kafka Server 0], Connecting to zookeeper on localhost:2181 (kafka.server.KafkaServer)
[2015-05-30 19:56:59,541] INFO Loading logs. (kafka.log.LogManager)
[2015-05-30 19:56:59,572] INFO Recovering unflushed segment 0 in log kafkatopic-0. (kafka.log.Log)
[2015-05-30 19:57:09,304] INFO Completed load of log kafkatopic-0 with log end offset 2065095 (kafka.log.Log)
[2015-05-30 19:57:09,304] INFO Recovering unflushed segment 0 in log my-repilcated-topic-0. (kafka.log.Log)
[2015-05-30 19:57:09,320] WARN Error when freeing index buffer (kafka.log.OffsetIndex)
java.lang.NullPointerException
	at kafka.log.OffsetIndex.kafka$log$OffsetIndex$$forceUnmap(OffsetIndex.scala:301)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:283)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:276)
	at kafka.utils.Utils$.inLock(Utils.scala:535)
	at kafka.log.OffsetIndex.resize(OffsetIndex.scala:276)
	at kafka.log.Log.loadSegments(Log.scala:179)
	at kafka.log.Log.<init>(Log.scala:67)
	at kafka.log.LogManager$$anonfun$loadLogs$2$$anonfun$3$$anonfun$apply$7$$anonfun$apply$1.apply$mcV$sp(LogManager.scala:142)
	at kafka.utils.Utils$$anon$1.run(Utils.scala:54)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:57:09,320] INFO Completed load of log my-repilcated-topic-0 with log end offset 19 (kafka.log.Log)
[2015-05-30 19:57:09,335] INFO Recovering unflushed segment 0 in log test-0. (kafka.log.Log)
[2015-05-30 19:57:09,335] WARN Error when freeing index buffer (kafka.log.OffsetIndex)
java.lang.NullPointerException
	at kafka.log.OffsetIndex.kafka$log$OffsetIndex$$forceUnmap(OffsetIndex.scala:301)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:283)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:276)
	at kafka.utils.Utils$.inLock(Utils.scala:535)
	at kafka.log.OffsetIndex.resize(OffsetIndex.scala:276)
	at kafka.log.Log.loadSegments(Log.scala:179)
	at kafka.log.Log.<init>(Log.scala:67)
	at kafka.log.LogManager$$anonfun$loadLogs$2$$anonfun$3$$anonfun$apply$7$$anonfun$apply$1.apply$mcV$sp(LogManager.scala:142)
	at kafka.utils.Utils$$anon$1.run(Utils.scala:54)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:57:09,335] INFO Completed load of log test-0 with log end offset 137 (kafka.log.Log)
[2015-05-30 19:57:09,335] INFO Logs loading complete. (kafka.log.LogManager)
[2015-05-30 19:57:09,335] INFO Starting log cleanup with a period of 300000 ms. (kafka.log.LogManager)
[2015-05-30 19:57:09,351] INFO Starting log flusher with a default period of 9223372036854775807 ms. (kafka.log.LogManager)
[2015-05-30 19:57:09,367] INFO Awaiting socket connections on 0.0.0.0:9092. (kafka.network.Acceptor)
[2015-05-30 19:57:09,367] INFO [Socket Server on Broker 0], Started (kafka.network.SocketServer)
[2015-05-30 19:57:09,414] INFO Will not load MX4J, mx4j-tools.jar is not in the classpath (kafka.utils.Mx4jLoader$)
[2015-05-30 19:57:09,445] INFO 0 successfully elected as leader (kafka.server.ZookeeperLeaderElector)
[2015-05-30 19:57:09,585] INFO New leader is 0 (kafka.server.ZookeeperLeaderElector$LeaderChangeListener)
[2015-05-30 19:57:09,601] INFO Registered broker 0 at path /brokers/ids/0 with address MININT-4IL6B3T.fareast.corp.microsoft.com:9092. (kafka.utils.ZkUtils$)
[2015-05-30 19:57:09,632] INFO [Kafka Server 0], started (kafka.server.KafkaServer)
[2015-05-30 19:57:09,942] INFO [ReplicaFetcherManager on broker 0] Removed fetcher for partitions [my-repilcated-topic,0],[test,0],[kafkatopic,0] (kafka.server.ReplicaFetcherManager)
[2015-05-30 19:57:10,073] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:57:10,157] INFO [ReplicaFetcherManager on broker 0] Removed fetcher for partitions [my-repilcated-topic,0],[test,0],[kafkatopic,0] (kafka.server.ReplicaFetcherManager)
[2015-05-30 19:57:10,462] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 19:57:24,150] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:57:34,028] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 19:57:35,698] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at kafka.utils.Utils$.read(Utils.scala:380)
	at kafka.network.BoundedByteBufferReceive.readFrom(BoundedByteBufferReceive.scala:54)
	at kafka.network.Processor.read(SocketServer.scala:444)
	at kafka.network.Processor.run(SocketServer.scala:340)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:58:01,709] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at kafka.utils.Utils$.read(Utils.scala:380)
	at kafka.network.BoundedByteBufferReceive.readFrom(BoundedByteBufferReceive.scala:54)
	at kafka.network.Processor.read(SocketServer.scala:444)
	at kafka.network.Processor.run(SocketServer.scala:340)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 19:59:23,482] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 19:59:41,065] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:229)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 20:00:54,919] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 20:01:08,842] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:229)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 20:01:22,256] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 20:01:42,946] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:229)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 20:02:49,449] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 20:02:57,621] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 20:02:57,621] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 20:03:34,375] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 20:04:13,610] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at kafka.utils.Utils$.read(Utils.scala:380)
	at kafka.network.BoundedByteBufferReceive.readFrom(BoundedByteBufferReceive.scala:54)
	at kafka.network.Processor.read(SocketServer.scala:444)
	at kafka.network.Processor.run(SocketServer.scala:340)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 20:04:21,206] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 20:04:52,641] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 20:05:11,178] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:229)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 20:05:16,233] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 20:05:22,607] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 20:05:40,135] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at kafka.utils.Utils$.read(Utils.scala:380)
	at kafka.network.BoundedByteBufferReceive.readFrom(BoundedByteBufferReceive.scala:54)
	at kafka.network.Processor.read(SocketServer.scala:444)
	at kafka.network.Processor.run(SocketServer.scala:340)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 20:06:12,567] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at kafka.utils.Utils$.read(Utils.scala:380)
	at kafka.network.BoundedByteBufferReceive.readFrom(BoundedByteBufferReceive.scala:54)
	at kafka.network.Processor.read(SocketServer.scala:444)
	at kafka.network.Processor.run(SocketServer.scala:340)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 20:06:48,961] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 20:07:45,077] INFO Verifying properties (kafka.utils.VerifiableProperties)
[2015-05-30 20:07:45,108] INFO Property broker.id is overridden to 0 (kafka.utils.VerifiableProperties)
[2015-05-30 20:07:45,108] INFO Property log.cleaner.enable is overridden to false (kafka.utils.VerifiableProperties)
[2015-05-30 20:07:45,108] INFO Property log.dirs is overridden to /tmp/kafka-logs (kafka.utils.VerifiableProperties)
[2015-05-30 20:07:45,108] INFO Property log.retention.check.interval.ms is overridden to 300000 (kafka.utils.VerifiableProperties)
[2015-05-30 20:07:45,108] INFO Property log.retention.hours is overridden to 168 (kafka.utils.VerifiableProperties)
[2015-05-30 20:07:45,108] INFO Property log.segment.bytes is overridden to 1073741824 (kafka.utils.VerifiableProperties)
[2015-05-30 20:07:45,108] INFO Property num.io.threads is overridden to 8 (kafka.utils.VerifiableProperties)
[2015-05-30 20:07:45,108] INFO Property num.network.threads is overridden to 3 (kafka.utils.VerifiableProperties)
[2015-05-30 20:07:45,108] INFO Property num.partitions is overridden to 1 (kafka.utils.VerifiableProperties)
[2015-05-30 20:07:45,108] INFO Property num.recovery.threads.per.data.dir is overridden to 1 (kafka.utils.VerifiableProperties)
[2015-05-30 20:07:45,108] INFO Property port is overridden to 9092 (kafka.utils.VerifiableProperties)
[2015-05-30 20:07:45,108] INFO Property socket.receive.buffer.bytes is overridden to 102400 (kafka.utils.VerifiableProperties)
[2015-05-30 20:07:45,108] INFO Property socket.request.max.bytes is overridden to 104857600 (kafka.utils.VerifiableProperties)
[2015-05-30 20:07:45,108] INFO Property socket.send.buffer.bytes is overridden to 102400 (kafka.utils.VerifiableProperties)
[2015-05-30 20:07:45,108] INFO Property zookeeper.connect is overridden to localhost:2181 (kafka.utils.VerifiableProperties)
[2015-05-30 20:07:45,108] INFO Property zookeeper.connection.timeout.ms is overridden to 6000 (kafka.utils.VerifiableProperties)
[2015-05-30 20:07:45,140] INFO [Kafka Server 0], starting (kafka.server.KafkaServer)
[2015-05-30 20:07:45,155] INFO [Kafka Server 0], Connecting to zookeeper on localhost:2181 (kafka.server.KafkaServer)
[2015-05-30 20:07:45,390] INFO Loading logs. (kafka.log.LogManager)
[2015-05-30 20:07:45,421] INFO Recovering unflushed segment 0 in log kafkatopic-0. (kafka.log.Log)
[2015-05-30 20:07:56,981] INFO Completed load of log kafkatopic-0 with log end offset 2420651 (kafka.log.Log)
[2015-05-30 20:07:56,981] INFO Recovering unflushed segment 0 in log my-repilcated-topic-0. (kafka.log.Log)
[2015-05-30 20:07:56,997] WARN Error when freeing index buffer (kafka.log.OffsetIndex)
java.lang.NullPointerException
	at kafka.log.OffsetIndex.kafka$log$OffsetIndex$$forceUnmap(OffsetIndex.scala:301)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:283)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:276)
	at kafka.utils.Utils$.inLock(Utils.scala:535)
	at kafka.log.OffsetIndex.resize(OffsetIndex.scala:276)
	at kafka.log.Log.loadSegments(Log.scala:179)
	at kafka.log.Log.<init>(Log.scala:67)
	at kafka.log.LogManager$$anonfun$loadLogs$2$$anonfun$3$$anonfun$apply$7$$anonfun$apply$1.apply$mcV$sp(LogManager.scala:142)
	at kafka.utils.Utils$$anon$1.run(Utils.scala:54)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 20:07:56,997] INFO Completed load of log my-repilcated-topic-0 with log end offset 19 (kafka.log.Log)
[2015-05-30 20:07:56,997] INFO Recovering unflushed segment 0 in log test-0. (kafka.log.Log)
[2015-05-30 20:07:57,012] WARN Error when freeing index buffer (kafka.log.OffsetIndex)
java.lang.NullPointerException
	at kafka.log.OffsetIndex.kafka$log$OffsetIndex$$forceUnmap(OffsetIndex.scala:301)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:283)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:276)
	at kafka.utils.Utils$.inLock(Utils.scala:535)
	at kafka.log.OffsetIndex.resize(OffsetIndex.scala:276)
	at kafka.log.Log.loadSegments(Log.scala:179)
	at kafka.log.Log.<init>(Log.scala:67)
	at kafka.log.LogManager$$anonfun$loadLogs$2$$anonfun$3$$anonfun$apply$7$$anonfun$apply$1.apply$mcV$sp(LogManager.scala:142)
	at kafka.utils.Utils$$anon$1.run(Utils.scala:54)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 20:07:57,012] INFO Completed load of log test-0 with log end offset 137 (kafka.log.Log)
[2015-05-30 20:07:57,012] INFO Logs loading complete. (kafka.log.LogManager)
[2015-05-30 20:07:57,012] INFO Starting log cleanup with a period of 300000 ms. (kafka.log.LogManager)
[2015-05-30 20:07:57,012] INFO Starting log flusher with a default period of 9223372036854775807 ms. (kafka.log.LogManager)
[2015-05-30 20:07:57,043] INFO Awaiting socket connections on 0.0.0.0:9092. (kafka.network.Acceptor)
[2015-05-30 20:07:57,043] INFO [Socket Server on Broker 0], Started (kafka.network.SocketServer)
[2015-05-30 20:07:57,090] INFO Will not load MX4J, mx4j-tools.jar is not in the classpath (kafka.utils.Mx4jLoader$)
[2015-05-30 20:07:57,106] INFO 0 successfully elected as leader (kafka.server.ZookeeperLeaderElector)
[2015-05-30 20:07:57,262] INFO New leader is 0 (kafka.server.ZookeeperLeaderElector$LeaderChangeListener)
[2015-05-30 20:07:57,262] INFO Registered broker 0 at path /brokers/ids/0 with address MININT-4IL6B3T.fareast.corp.microsoft.com:9092. (kafka.utils.ZkUtils$)
[2015-05-30 20:07:57,278] INFO [Kafka Server 0], started (kafka.server.KafkaServer)
[2015-05-30 20:07:57,418] INFO [ReplicaFetcherManager on broker 0] Removed fetcher for partitions [my-repilcated-topic,0],[test,0],[kafkatopic,0] (kafka.server.ReplicaFetcherManager)
[2015-05-30 20:07:57,450] INFO [ReplicaFetcherManager on broker 0] Removed fetcher for partitions [my-repilcated-topic,0],[test,0],[kafkatopic,0] (kafka.server.ReplicaFetcherManager)
[2015-05-30 20:08:41,368] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 20:08:45,823] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 20:09:18,658] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at kafka.utils.Utils$.read(Utils.scala:380)
	at kafka.network.BoundedByteBufferReceive.readFrom(BoundedByteBufferReceive.scala:54)
	at kafka.network.Processor.read(SocketServer.scala:444)
	at kafka.network.Processor.run(SocketServer.scala:340)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 20:09:21,601] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 20:09:30,925] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at kafka.utils.Utils$.read(Utils.scala:380)
	at kafka.network.BoundedByteBufferReceive.readFrom(BoundedByteBufferReceive.scala:54)
	at kafka.network.Processor.read(SocketServer.scala:444)
	at kafka.network.Processor.run(SocketServer.scala:340)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 20:09:39,862] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 20:09:48,245] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 20:10:01,937] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at kafka.utils.Utils$.read(Utils.scala:380)
	at kafka.network.BoundedByteBufferReceive.readFrom(BoundedByteBufferReceive.scala:54)
	at kafka.network.Processor.read(SocketServer.scala:444)
	at kafka.network.Processor.run(SocketServer.scala:340)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 20:10:45,216] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 20:11:07,692] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 20:11:07,707] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An established connection was aborted by the software in your host machine
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.TopicDataSend.writeTo(FetchResponse.scala:123)
	at kafka.network.MultiSend.writeTo(Transmission.scala:101)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:231)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 20:11:12,844] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An established connection was aborted by the software in your host machine
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.TopicDataSend.writeTo(FetchResponse.scala:123)
	at kafka.network.MultiSend.writeTo(Transmission.scala:101)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:231)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 20:11:12,875] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 20:11:15,094] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An established connection was aborted by the software in your host machine
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.TopicDataSend.writeTo(FetchResponse.scala:123)
	at kafka.network.MultiSend.writeTo(Transmission.scala:101)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:231)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 20:11:15,188] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 20:11:33,224] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:229)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 20:12:09,272] INFO Verifying properties (kafka.utils.VerifiableProperties)
[2015-05-30 20:12:09,319] INFO Property broker.id is overridden to 0 (kafka.utils.VerifiableProperties)
[2015-05-30 20:12:09,319] INFO Property log.cleaner.enable is overridden to false (kafka.utils.VerifiableProperties)
[2015-05-30 20:12:09,319] INFO Property log.dirs is overridden to /tmp/kafka-logs (kafka.utils.VerifiableProperties)
[2015-05-30 20:12:09,319] INFO Property log.retention.check.interval.ms is overridden to 300000 (kafka.utils.VerifiableProperties)
[2015-05-30 20:12:09,319] INFO Property log.retention.hours is overridden to 168 (kafka.utils.VerifiableProperties)
[2015-05-30 20:12:09,319] INFO Property log.segment.bytes is overridden to 1073741824 (kafka.utils.VerifiableProperties)
[2015-05-30 20:12:09,319] INFO Property num.io.threads is overridden to 8 (kafka.utils.VerifiableProperties)
[2015-05-30 20:12:09,319] INFO Property num.network.threads is overridden to 3 (kafka.utils.VerifiableProperties)
[2015-05-30 20:12:09,319] INFO Property num.partitions is overridden to 1 (kafka.utils.VerifiableProperties)
[2015-05-30 20:12:09,319] INFO Property num.recovery.threads.per.data.dir is overridden to 1 (kafka.utils.VerifiableProperties)
[2015-05-30 20:12:09,319] INFO Property port is overridden to 9092 (kafka.utils.VerifiableProperties)
[2015-05-30 20:12:09,319] INFO Property socket.receive.buffer.bytes is overridden to 102400 (kafka.utils.VerifiableProperties)
[2015-05-30 20:12:09,319] INFO Property socket.request.max.bytes is overridden to 104857600 (kafka.utils.VerifiableProperties)
[2015-05-30 20:12:09,319] INFO Property socket.send.buffer.bytes is overridden to 102400 (kafka.utils.VerifiableProperties)
[2015-05-30 20:12:09,319] INFO Property zookeeper.connect is overridden to localhost:2181 (kafka.utils.VerifiableProperties)
[2015-05-30 20:12:09,319] INFO Property zookeeper.connection.timeout.ms is overridden to 6000 (kafka.utils.VerifiableProperties)
[2015-05-30 20:12:09,366] INFO [Kafka Server 0], starting (kafka.server.KafkaServer)
[2015-05-30 20:12:09,366] INFO [Kafka Server 0], Connecting to zookeeper on localhost:2181 (kafka.server.KafkaServer)
[2015-05-30 20:12:09,710] INFO Loading logs. (kafka.log.LogManager)
[2015-05-30 20:12:09,741] INFO Recovering unflushed segment 0 in log kafkatopic-0. (kafka.log.Log)
[2015-05-30 20:12:35,611] INFO Completed load of log kafkatopic-0 with log end offset 2767465 (kafka.log.Log)
[2015-05-30 20:12:35,661] INFO Recovering unflushed segment 0 in log my-repilcated-topic-0. (kafka.log.Log)
[2015-05-30 20:12:35,671] WARN Error when freeing index buffer (kafka.log.OffsetIndex)
java.lang.NullPointerException
	at kafka.log.OffsetIndex.kafka$log$OffsetIndex$$forceUnmap(OffsetIndex.scala:301)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:283)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:276)
	at kafka.utils.Utils$.inLock(Utils.scala:535)
	at kafka.log.OffsetIndex.resize(OffsetIndex.scala:276)
	at kafka.log.Log.loadSegments(Log.scala:179)
	at kafka.log.Log.<init>(Log.scala:67)
	at kafka.log.LogManager$$anonfun$loadLogs$2$$anonfun$3$$anonfun$apply$7$$anonfun$apply$1.apply$mcV$sp(LogManager.scala:142)
	at kafka.utils.Utils$$anon$1.run(Utils.scala:54)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 20:12:35,679] INFO Completed load of log my-repilcated-topic-0 with log end offset 19 (kafka.log.Log)
[2015-05-30 20:12:35,695] INFO Recovering unflushed segment 0 in log test-0. (kafka.log.Log)
[2015-05-30 20:12:35,706] WARN Error when freeing index buffer (kafka.log.OffsetIndex)
java.lang.NullPointerException
	at kafka.log.OffsetIndex.kafka$log$OffsetIndex$$forceUnmap(OffsetIndex.scala:301)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:283)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:276)
	at kafka.utils.Utils$.inLock(Utils.scala:535)
	at kafka.log.OffsetIndex.resize(OffsetIndex.scala:276)
	at kafka.log.Log.loadSegments(Log.scala:179)
	at kafka.log.Log.<init>(Log.scala:67)
	at kafka.log.LogManager$$anonfun$loadLogs$2$$anonfun$3$$anonfun$apply$7$$anonfun$apply$1.apply$mcV$sp(LogManager.scala:142)
	at kafka.utils.Utils$$anon$1.run(Utils.scala:54)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 20:12:35,715] INFO Completed load of log test-0 with log end offset 137 (kafka.log.Log)
[2015-05-30 20:12:35,738] INFO Logs loading complete. (kafka.log.LogManager)
[2015-05-30 20:12:35,740] INFO Starting log cleanup with a period of 300000 ms. (kafka.log.LogManager)
[2015-05-30 20:12:35,751] INFO Starting log flusher with a default period of 9223372036854775807 ms. (kafka.log.LogManager)
[2015-05-30 20:12:35,803] INFO Awaiting socket connections on 0.0.0.0:9092. (kafka.network.Acceptor)
[2015-05-30 20:12:35,809] INFO [Socket Server on Broker 0], Started (kafka.network.SocketServer)
[2015-05-30 20:12:35,892] INFO Will not load MX4J, mx4j-tools.jar is not in the classpath (kafka.utils.Mx4jLoader$)
[2015-05-30 20:12:35,941] INFO 0 successfully elected as leader (kafka.server.ZookeeperLeaderElector)
[2015-05-30 20:12:36,239] INFO New leader is 0 (kafka.server.ZookeeperLeaderElector$LeaderChangeListener)
[2015-05-30 20:12:36,266] INFO Registered broker 0 at path /brokers/ids/0 with address MININT-4IL6B3T.fareast.corp.microsoft.com:9092. (kafka.utils.ZkUtils$)
[2015-05-30 20:12:36,368] INFO [Kafka Server 0], started (kafka.server.KafkaServer)
[2015-05-30 20:12:36,571] INFO [ReplicaFetcherManager on broker 0] Removed fetcher for partitions [my-repilcated-topic,0],[test,0],[kafkatopic,0] (kafka.server.ReplicaFetcherManager)
[2015-05-30 20:12:36,647] INFO [ReplicaFetcherManager on broker 0] Removed fetcher for partitions [my-repilcated-topic,0],[test,0],[kafkatopic,0] (kafka.server.ReplicaFetcherManager)
[2015-05-30 20:12:36,675] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 20:13:16,846] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 20:13:24,420] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 20:13:46,103] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:229)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 20:13:51,192] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 20:14:10,423] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:229)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 20:14:54,085] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 20:15:07,861] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:229)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 20:18:23,995] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 20:18:37,514] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 20:18:37,514] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 20:18:39,726] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at kafka.utils.Utils$.read(Utils.scala:380)
	at kafka.network.BoundedByteBufferReceive.readFrom(BoundedByteBufferReceive.scala:54)
	at kafka.network.Processor.read(SocketServer.scala:444)
	at kafka.network.Processor.run(SocketServer.scala:340)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 20:18:41,217] ERROR Closing socket for /10.86.164.32 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:197)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at kafka.utils.Utils$.read(Utils.scala:380)
	at kafka.network.BoundedByteBufferReceive.readFrom(BoundedByteBufferReceive.scala:54)
	at kafka.network.Processor.read(SocketServer.scala:444)
	at kafka.network.Processor.run(SocketServer.scala:340)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 20:19:32,908] INFO Verifying properties (kafka.utils.VerifiableProperties)
[2015-05-30 20:19:32,940] INFO Property broker.id is overridden to 0 (kafka.utils.VerifiableProperties)
[2015-05-30 20:19:32,940] INFO Property log.cleaner.enable is overridden to false (kafka.utils.VerifiableProperties)
[2015-05-30 20:19:32,940] INFO Property log.dirs is overridden to /tmp/kafka-logs (kafka.utils.VerifiableProperties)
[2015-05-30 20:19:32,940] INFO Property log.retention.check.interval.ms is overridden to 300000 (kafka.utils.VerifiableProperties)
[2015-05-30 20:19:32,940] INFO Property log.retention.hours is overridden to 168 (kafka.utils.VerifiableProperties)
[2015-05-30 20:19:32,940] INFO Property log.segment.bytes is overridden to 1073741824 (kafka.utils.VerifiableProperties)
[2015-05-30 20:19:32,940] INFO Property num.io.threads is overridden to 8 (kafka.utils.VerifiableProperties)
[2015-05-30 20:19:32,940] INFO Property num.network.threads is overridden to 3 (kafka.utils.VerifiableProperties)
[2015-05-30 20:19:32,940] INFO Property num.partitions is overridden to 1 (kafka.utils.VerifiableProperties)
[2015-05-30 20:19:32,940] INFO Property num.recovery.threads.per.data.dir is overridden to 1 (kafka.utils.VerifiableProperties)
[2015-05-30 20:19:32,940] INFO Property port is overridden to 9092 (kafka.utils.VerifiableProperties)
[2015-05-30 20:19:32,940] INFO Property socket.receive.buffer.bytes is overridden to 102400 (kafka.utils.VerifiableProperties)
[2015-05-30 20:19:32,940] INFO Property socket.request.max.bytes is overridden to 104857600 (kafka.utils.VerifiableProperties)
[2015-05-30 20:19:32,940] INFO Property socket.send.buffer.bytes is overridden to 102400 (kafka.utils.VerifiableProperties)
[2015-05-30 20:19:32,940] INFO Property zookeeper.connect is overridden to localhost:2181 (kafka.utils.VerifiableProperties)
[2015-05-30 20:19:32,940] INFO Property zookeeper.connection.timeout.ms is overridden to 6000 (kafka.utils.VerifiableProperties)
[2015-05-30 20:19:32,987] INFO [Kafka Server 0], starting (kafka.server.KafkaServer)
[2015-05-30 20:19:33,002] INFO [Kafka Server 0], Connecting to zookeeper on localhost:2181 (kafka.server.KafkaServer)
[2015-05-30 20:19:33,252] INFO Loading logs. (kafka.log.LogManager)
[2015-05-30 20:19:33,283] INFO Recovering unflushed segment 0 in log kafkatopic-0. (kafka.log.Log)
[2015-05-30 20:19:47,220] INFO Completed load of log kafkatopic-0 with log end offset 2830588 (kafka.log.Log)
[2015-05-30 20:19:47,236] INFO Recovering unflushed segment 0 in log my-repilcated-topic-0. (kafka.log.Log)
[2015-05-30 20:19:47,236] WARN Error when freeing index buffer (kafka.log.OffsetIndex)
java.lang.NullPointerException
	at kafka.log.OffsetIndex.kafka$log$OffsetIndex$$forceUnmap(OffsetIndex.scala:301)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:283)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:276)
	at kafka.utils.Utils$.inLock(Utils.scala:535)
	at kafka.log.OffsetIndex.resize(OffsetIndex.scala:276)
	at kafka.log.Log.loadSegments(Log.scala:179)
	at kafka.log.Log.<init>(Log.scala:67)
	at kafka.log.LogManager$$anonfun$loadLogs$2$$anonfun$3$$anonfun$apply$7$$anonfun$apply$1.apply$mcV$sp(LogManager.scala:142)
	at kafka.utils.Utils$$anon$1.run(Utils.scala:54)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 20:19:47,251] INFO Completed load of log my-repilcated-topic-0 with log end offset 19 (kafka.log.Log)
[2015-05-30 20:19:47,251] INFO Recovering unflushed segment 0 in log test-0. (kafka.log.Log)
[2015-05-30 20:19:47,267] WARN Error when freeing index buffer (kafka.log.OffsetIndex)
java.lang.NullPointerException
	at kafka.log.OffsetIndex.kafka$log$OffsetIndex$$forceUnmap(OffsetIndex.scala:301)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:283)
	at kafka.log.OffsetIndex$$anonfun$resize$1.apply(OffsetIndex.scala:276)
	at kafka.utils.Utils$.inLock(Utils.scala:535)
	at kafka.log.OffsetIndex.resize(OffsetIndex.scala:276)
	at kafka.log.Log.loadSegments(Log.scala:179)
	at kafka.log.Log.<init>(Log.scala:67)
	at kafka.log.LogManager$$anonfun$loadLogs$2$$anonfun$3$$anonfun$apply$7$$anonfun$apply$1.apply$mcV$sp(LogManager.scala:142)
	at kafka.utils.Utils$$anon$1.run(Utils.scala:54)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 20:19:47,267] INFO Completed load of log test-0 with log end offset 137 (kafka.log.Log)
[2015-05-30 20:19:47,267] INFO Logs loading complete. (kafka.log.LogManager)
[2015-05-30 20:19:47,283] INFO Starting log cleanup with a period of 300000 ms. (kafka.log.LogManager)
[2015-05-30 20:19:47,283] INFO Starting log flusher with a default period of 9223372036854775807 ms. (kafka.log.LogManager)
[2015-05-30 20:19:47,298] INFO Awaiting socket connections on 0.0.0.0:9092. (kafka.network.Acceptor)
[2015-05-30 20:19:47,298] INFO [Socket Server on Broker 0], Started (kafka.network.SocketServer)
[2015-05-30 20:19:47,345] INFO Will not load MX4J, mx4j-tools.jar is not in the classpath (kafka.utils.Mx4jLoader$)
[2015-05-30 20:19:47,376] INFO 0 successfully elected as leader (kafka.server.ZookeeperLeaderElector)
[2015-05-30 20:19:47,533] INFO New leader is 0 (kafka.server.ZookeeperLeaderElector$LeaderChangeListener)
[2015-05-30 20:19:47,533] INFO Registered broker 0 at path /brokers/ids/0 with address MININT-4IL6B3T.fareast.corp.microsoft.com:9092. (kafka.utils.ZkUtils$)
[2015-05-30 20:19:47,548] INFO [Kafka Server 0], started (kafka.server.KafkaServer)
[2015-05-30 20:19:47,705] INFO [ReplicaFetcherManager on broker 0] Removed fetcher for partitions [my-repilcated-topic,0],[test,0],[kafkatopic,0] (kafka.server.ReplicaFetcherManager)
[2015-05-30 20:19:47,736] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 20:19:47,783] WARN [KafkaApi-0] Offset request with correlation id 0 from client console-consumer-50547 on partition [kafkatopic,0] failed due to Leader not local for partition [kafkatopic,0] on broker 0 (kafka.server.KafkaApis)
[2015-05-30 20:19:47,798] INFO [ReplicaFetcherManager on broker 0] Removed fetcher for partitions [my-repilcated-topic,0],[test,0],[kafkatopic,0] (kafka.server.ReplicaFetcherManager)
[2015-05-30 20:19:47,814] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 20:19:48,033] INFO Closing socket connection to /10.86.164.32. (kafka.network.Processor)
[2015-05-30 20:20:27,920] INFO Closing socket connection to /127.0.0.1. (kafka.network.Processor)
[2015-05-30 20:20:38,778] INFO Closing socket connection to /10.172.120.154. (kafka.network.Processor)
[2015-05-30 20:21:41,764] ERROR Closing socket for /10.172.120.154 because of error (kafka.network.Processor)
java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.write0(Native Method)
	at sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:51)
	at sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:93)
	at sun.nio.ch.IOUtil.write(IOUtil.java:65)
	at sun.nio.ch.SocketChannelImpl.write(SocketChannelImpl.java:471)
	at kafka.api.FetchResponseSend.writeTo(FetchResponse.scala:229)
	at kafka.network.Processor.write(SocketServer.scala:472)
	at kafka.network.Processor.run(SocketServer.scala:342)
	at java.lang.Thread.run(Thread.java:745)
[2015-05-30 20:21:57,562] INFO Closing socket connection to /10.172.120.154. (kafka.network.Processor)
